{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "(cmdb) [~/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics]tar -xvf week13_data.tar "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "conda install krona\n",
    "ktImportText -q SRR4921*"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "(cmdb) [~/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics]conda install metabat2\n",
    "(cmdb) [~/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics]tar -xvf bins.tar"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "file = open('/Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/KRAKEN/SRR492183.kraken', 'r')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'kraken_83' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-4-9363da145032>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mTEMPDIC\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mx\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mkraken_83\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      6\u001b[0m     \u001b[0mtemp\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'\\t'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'kraken_83' is not defined"
     ]
    }
   ],
   "source": [
    "Filelist = ['83','86','88','89','90','93','94','97']\n",
    "\n",
    "TEMPDIC = {}\n",
    "for x in kraken_83:\n",
    "    temp = x.split('\\t')\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def dict_maker(num):\n",
    "    f = open(\"/Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/KRAKEN/SRR4921\"+num+\".kraken\")\n",
    "    kraken = f.readlines()\n",
    "    f.close()\n",
    "    day_temp = {}\n",
    "    for line in kraken:\n",
    "        key = line.split('\\t')[1]\n",
    "        day_temp.setdefault(key,0)\n",
    "        day_temp[key] += 1\n",
    "    f = open(\"/Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/KRAKEN/SRR4921day\"+num+\".kraken\",'w')\n",
    "    for key,val in day_temp.items():\n",
    "        x = key.split(';')\n",
    "        f.write(str(val) + '\\t')\n",
    "        for y in x[:-1]:\n",
    "            f.write(y+'\\t')\n",
    "        f.write(x[-1])\n",
    "    f.close()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "Filelist = ['83','86','88','89','90','93','94','97']\n",
    "for number in Filelist:\n",
    "    dict_maker(number)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing week13_data/KRAKEN/SRR4921day83.krona.html...\n",
      "Writing week13_data/KRAKEN/SRR4921day86.krona.html...\n",
      "Writing week13_data/KRAKEN/SRR4921day88.krona.html...\n",
      "Writing week13_data/KRAKEN/SRR4921day89.krona.html...\n",
      "Writing week13_data/KRAKEN/SRR4921day90.krona.html...\n",
      "Writing week13_data/KRAKEN/SRR4921day93.krona.html...\n",
      "Writing week13_data/KRAKEN/SRR4921day94.krona.html...\n",
      "Writing week13_data/KRAKEN/SRR4921day97.krona.html...\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "No such file or directory at /Users/cmdb/miniconda3/envs/cmdb/bin/ktImportText line 71.\n",
      "No such file or directory at /Users/cmdb/miniconda3/envs/cmdb/bin/ktImportText line 71.\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "for end in { 83 86 88 89 90 93 94 97 }\n",
    "do\n",
    "ktImportText -o week13_data/KRAKEN/SRR4921day${end}.krona.html week13_data/KRAKEN/SRR4921day${end}.kraken\n",
    "done"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "bwa index /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/assembly.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "bwa mem /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/assembly.fasta /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492183_1.fastq /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492183_2.fastq > test.sam"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "#%%bash\n",
    "#chmod +x BASHbwa_mem.ipynb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 127788 sequences (10000033 bp)...\n",
      "[M::process] read 125606 sequences (10000105 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 55486, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (901, 942, 995)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (713, 1183)\n",
      "[M::mem_pestat] mean and std.dev: (951.07, 70.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (619, 1277)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 127788 reads in 6.771 CPU sec, 7.612 real sec\n",
      "[M::process] read 124886 sequences (10000101 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 55068, 13, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (903, 944, 997)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (715, 1185)\n",
      "[M::mem_pestat] mean and std.dev: (953.18, 70.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (621, 1279)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (28, 401, 446)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1282)\n",
      "[M::mem_pestat] mean and std.dev: (212.36, 197.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1700)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 125606 reads in 6.452 CPU sec, 6.536 real sec\n",
      "[M::process] read 125478 sequences (10000158 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 54958, 11, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (903, 945, 997)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (715, 1185)\n",
      "[M::mem_pestat] mean and std.dev: (953.40, 70.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (621, 1279)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (150, 349, 814)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 2142)\n",
      "[M::mem_pestat] mean and std.dev: (414.09, 396.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2806)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 124886 reads in 6.760 CPU sec, 8.096 real sec\n",
      "[M::process] read 124244 sequences (10000139 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 55104, 13, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (903, 945, 998)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (713, 1188)\n",
      "[M::mem_pestat] mean and std.dev: (953.93, 70.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (618, 1283)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (349, 469, 1296)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3190)\n",
      "[M::mem_pestat] mean and std.dev: (733.08, 523.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 4137)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 125478 reads in 6.632 CPU sec, 7.109 real sec\n",
      "[M::process] read 124170 sequences (10000108 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 54929, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (903, 945, 998)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (713, 1188)\n",
      "[M::mem_pestat] mean and std.dev: (953.45, 70.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (618, 1283)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 124244 reads in 6.603 CPU sec, 7.009 real sec\n",
      "[M::process] read 123612 sequences (10000026 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 55005, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (904, 945, 999)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (714, 1189)\n",
      "[M::mem_pestat] mean and std.dev: (954.38, 69.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (619, 1284)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 124170 reads in 6.399 CPU sec, 6.438 real sec\n",
      "[M::process] read 123020 sequences (10000010 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 55184, 5, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (904, 946, 1000)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (712, 1192)\n",
      "[M::mem_pestat] mean and std.dev: (954.30, 70.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (616, 1288)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 123612 reads in 6.447 CPU sec, 6.509 real sec\n",
      "[M::process] read 121610 sequences (10000085 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 55045, 2, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (904, 946, 999)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (714, 1189)\n",
      "[M::mem_pestat] mean and std.dev: (954.78, 69.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (619, 1284)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 123020 reads in 6.441 CPU sec, 6.525 real sec\n",
      "[M::process] read 121416 sequences (10000188 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 54521, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (905, 946, 1000)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (715, 1190)\n",
      "[M::mem_pestat] mean and std.dev: (954.89, 69.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (620, 1285)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 121610 reads in 6.412 CPU sec, 6.576 real sec\n",
      "[M::process] read 121510 sequences (10000095 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 54474, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (905, 945, 1000)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (715, 1190)\n",
      "[M::mem_pestat] mean and std.dev: (955.16, 70.31)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (620, 1285)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 121416 reads in 6.313 CPU sec, 6.384 real sec\n",
      "[M::process] read 121334 sequences (10000077 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54626, 5, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (904, 946, 999)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (714, 1189)\n",
      "[M::mem_pestat] mean and std.dev: (954.85, 70.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (619, 1284)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 121510 reads in 6.222 CPU sec, 6.247 real sec\n",
      "[M::process] read 121378 sequences (10000114 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54522, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (904, 946, 999)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (714, 1189)\n",
      "[M::mem_pestat] mean and std.dev: (954.85, 70.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (619, 1284)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 121334 reads in 6.243 CPU sec, 6.262 real sec\n",
      "[M::process] read 122522 sequences (10000036 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 54537, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (904, 945, 999)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (714, 1189)\n",
      "[M::mem_pestat] mean and std.dev: (954.45, 70.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (619, 1284)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 121378 reads in 6.246 CPU sec, 6.412 real sec\n",
      "[M::process] read 121196 sequences (10000005 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54868, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (904, 945, 999)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (714, 1189)\n",
      "[M::mem_pestat] mean and std.dev: (954.37, 69.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (619, 1284)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 122522 reads in 6.291 CPU sec, 6.447 real sec\n",
      "[M::process] read 118166 sequences (10000110 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54528, 12, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (904, 946, 1000)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (712, 1192)\n",
      "[M::mem_pestat] mean and std.dev: (955.08, 70.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (616, 1288)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (278, 278, 455)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 809)\n",
      "[M::mem_pestat] mean and std.dev: (261.20, 158.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 986)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 121196 reads in 6.296 CPU sec, 6.405 real sec\n",
      "[M::process] read 32064 sequences (2721990 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53808, 2, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (904, 947, 1001)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (710, 1195)\n",
      "[M::mem_pestat] mean and std.dev: (955.39, 71.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (613, 1292)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 118166 reads in 7.848 CPU sec, 8.071 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 14477, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (905, 946, 999)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (717, 1187)\n",
      "[M::mem_pestat] mean and std.dev: (954.82, 70.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (623, 1281)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 32064 reads in 1.938 CPU sec, 1.831 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/assembly.fasta /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492183_1.fastq /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492183_2.fastq\n",
      "[main] Real time: 110.720 sec; CPU: 106.545 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 117168 sequences (10000179 bp)...\n",
      "[M::process] read 116476 sequences (10000127 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 54963, 5, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (823, 862, 913)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (643, 1093)\n",
      "[M::mem_pestat] mean and std.dev: (869.11, 61.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (553, 1183)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 117168 reads in 5.955 CPU sec, 6.355 real sec\n",
      "[M::process] read 115200 sequences (10000116 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 54720, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (824, 864, 915)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (642, 1097)\n",
      "[M::mem_pestat] mean and std.dev: (870.12, 62.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (551, 1188)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 116476 reads in 5.837 CPU sec, 5.955 real sec\n",
      "[M::process] read 115010 sequences (10000144 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 54308, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (824, 864, 915)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (642, 1097)\n",
      "[M::mem_pestat] mean and std.dev: (870.42, 62.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (551, 1188)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 115200 reads in 6.037 CPU sec, 6.339 real sec\n",
      "[M::process] read 114712 sequences (10000081 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 54270, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (824, 865, 915)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (642, 1097)\n",
      "[M::mem_pestat] mean and std.dev: (870.86, 61.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (551, 1188)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 115010 reads in 5.860 CPU sec, 5.975 real sec\n",
      "[M::process] read 114396 sequences (10000126 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54220, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (824, 865, 914)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (644, 1094)\n",
      "[M::mem_pestat] mean and std.dev: (870.54, 61.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (554, 1184)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114712 reads in 5.850 CPU sec, 5.867 real sec\n",
      "[M::process] read 114650 sequences (10000073 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 54202, 9, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (823, 863, 915)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (639, 1099)\n",
      "[M::mem_pestat] mean and std.dev: (869.85, 62.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (547, 1191)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114396 reads in 5.782 CPU sec, 5.904 real sec\n",
      "[M::process] read 114156 sequences (10000114 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54269, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (825, 864, 915)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (645, 1095)\n",
      "[M::mem_pestat] mean and std.dev: (871.00, 61.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (555, 1185)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114650 reads in 5.796 CPU sec, 5.872 real sec\n",
      "[M::process] read 113964 sequences (10000090 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 54124, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (824, 865, 915)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (642, 1097)\n",
      "[M::mem_pestat] mean and std.dev: (870.70, 62.20)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (551, 1188)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114156 reads in 5.785 CPU sec, 5.829 real sec\n",
      "[M::process] read 113676 sequences (10000157 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54065, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (824, 864, 915)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (642, 1097)\n",
      "[M::mem_pestat] mean and std.dev: (870.48, 61.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (551, 1188)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 113964 reads in 5.708 CPU sec, 5.660 real sec\n",
      "[M::process] read 113646 sequences (10000058 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 53984, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (824, 864, 914)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (644, 1094)\n",
      "[M::mem_pestat] mean and std.dev: (870.22, 61.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (554, 1184)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 113676 reads in 5.723 CPU sec, 5.756 real sec\n",
      "[M::process] read 113840 sequences (10000094 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 54035, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (825, 865, 915)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (645, 1095)\n",
      "[M::mem_pestat] mean and std.dev: (871.00, 62.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (555, 1185)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 113646 reads in 5.679 CPU sec, 5.690 real sec\n",
      "[M::process] read 113366 sequences (10000053 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54237, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (825, 865, 917)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (641, 1101)\n",
      "[M::mem_pestat] mean and std.dev: (871.29, 62.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (549, 1193)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 113840 reads in 5.788 CPU sec, 5.819 real sec\n",
      "[M::process] read 112994 sequences (10000185 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53974, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (824, 865, 916)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (640, 1100)\n",
      "[M::mem_pestat] mean and std.dev: (871.04, 61.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (548, 1192)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 113366 reads in 5.812 CPU sec, 5.829 real sec\n",
      "[M::process] read 112740 sequences (10000088 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53900, 2, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (824, 865, 916)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (640, 1100)\n",
      "[M::mem_pestat] mean and std.dev: (870.82, 62.40)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (548, 1192)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 112994 reads in 5.906 CPU sec, 6.006 real sec\n",
      "[M::process] read 111964 sequences (10000160 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53700, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (824, 865, 916)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (640, 1100)\n",
      "[M::mem_pestat] mean and std.dev: (871.04, 61.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (548, 1192)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 112740 reads in 5.813 CPU sec, 5.894 real sec\n",
      "[M::process] read 111654 sequences (10000164 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53517, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (824, 864, 916)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (640, 1100)\n",
      "[M::mem_pestat] mean and std.dev: (870.59, 62.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (548, 1192)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 111964 reads in 6.212 CPU sec, 7.154 real sec\n",
      "[M::process] read 60388 sequences (5423467 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53341, 7, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (824, 864, 916)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (640, 1100)\n",
      "[M::mem_pestat] mean and std.dev: (870.39, 62.11)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (548, 1192)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 111654 reads in 6.295 CPU sec, 7.977 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 28865, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (825, 865, 916)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (643, 1098)\n",
      "[M::mem_pestat] mean and std.dev: (871.31, 61.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (552, 1189)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 60388 reads in 5.512 CPU sec, 5.798 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/assembly.fasta /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492186_1.fastq /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492186_2.fastq\n",
      "[main] Real time: 110.098 sec; CPU: 105.672 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 103464 sequences (10000093 bp)...\n",
      "[M::process] read 103672 sequences (10000185 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 48652, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (832, 871, 920)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (656, 1096)\n",
      "[M::mem_pestat] mean and std.dev: (877.15, 64.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (568, 1184)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 103464 reads in 6.030 CPU sec, 6.408 real sec\n",
      "[M::process] read 102822 sequences (10000091 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48796, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (832, 870, 918)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (660, 1090)\n",
      "[M::mem_pestat] mean and std.dev: (876.61, 63.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (574, 1176)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 103672 reads in 5.929 CPU sec, 5.982 real sec\n",
      "[M::process] read 104156 sequences (10000190 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48632, 7, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 871, 919)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (661, 1091)\n",
      "[M::mem_pestat] mean and std.dev: (877.01, 63.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (575, 1177)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 102822 reads in 5.660 CPU sec, 5.707 real sec\n",
      "[M::process] read 102860 sequences (10000030 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 49194, 3, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (832, 870, 919)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (658, 1093)\n",
      "[M::mem_pestat] mean and std.dev: (876.70, 63.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (571, 1180)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104156 reads in 5.814 CPU sec, 5.870 real sec\n",
      "[M::process] read 104194 sequences (10000023 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48789, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 871, 920)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (659, 1094)\n",
      "[M::mem_pestat] mean and std.dev: (877.17, 63.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (572, 1181)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 102860 reads in 5.850 CPU sec, 6.051 real sec\n",
      "[M::process] read 104050 sequences (10000188 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 49027, 5, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 870, 919)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (661, 1091)\n",
      "[M::mem_pestat] mean and std.dev: (876.68, 63.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (575, 1177)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104194 reads in 8.112 CPU sec, 11.347 real sec\n",
      "[M::process] read 103038 sequences (10000060 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48786, 0, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 871, 920)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (659, 1094)\n",
      "[M::mem_pestat] mean and std.dev: (877.42, 63.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (572, 1181)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104050 reads in 9.158 CPU sec, 12.997 real sec\n",
      "[M::process] read 104040 sequences (10000029 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48871, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 871, 920)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (659, 1094)\n",
      "[M::mem_pestat] mean and std.dev: (877.32, 63.97)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (572, 1181)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 103038 reads in 6.771 CPU sec, 7.636 real sec\n",
      "[M::process] read 102834 sequences (10000056 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 49081, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 871, 920)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (659, 1094)\n",
      "[M::mem_pestat] mean and std.dev: (877.21, 63.62)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (572, 1181)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104040 reads in 7.314 CPU sec, 12.306 real sec\n",
      "[M::process] read 103960 sequences (10000005 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48563, 2, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (832, 870, 919)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (658, 1093)\n",
      "[M::mem_pestat] mean and std.dev: (876.63, 63.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (571, 1180)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 102834 reads in 7.368 CPU sec, 10.530 real sec\n",
      "[M::process] read 103724 sequences (10000025 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48911, 5, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 870, 920)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (659, 1094)\n",
      "[M::mem_pestat] mean and std.dev: (876.84, 63.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (572, 1181)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 103960 reads in 6.781 CPU sec, 8.502 real sec\n",
      "[M::process] read 103574 sequences (10000033 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48711, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 871, 920)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (659, 1094)\n",
      "[M::mem_pestat] mean and std.dev: (876.79, 63.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (572, 1181)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 103724 reads in 6.388 CPU sec, 6.596 real sec\n",
      "[M::process] read 104390 sequences (10000045 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 49054, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 870, 919)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (661, 1091)\n",
      "[M::mem_pestat] mean and std.dev: (876.97, 63.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (575, 1177)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 103574 reads in 5.984 CPU sec, 5.992 real sec\n",
      "[M::process] read 102718 sequences (10000161 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 49379, 10, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (832, 871, 920)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (656, 1096)\n",
      "[M::mem_pestat] mean and std.dev: (876.81, 64.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (568, 1184)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (31, 140, 4256)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 12706)\n",
      "[M::mem_pestat] mean and std.dev: (1701.80, 2674.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 16931)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 104390 reads in 6.306 CPU sec, 6.784 real sec\n",
      "[M::process] read 104196 sequences (10000004 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48528, 1, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 870, 919)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (661, 1091)\n",
      "[M::mem_pestat] mean and std.dev: (876.86, 63.64)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (575, 1177)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 102718 reads in 6.425 CPU sec, 6.946 real sec\n",
      "[M::process] read 103964 sequences (10000099 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 49256, 10, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 872, 920)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (659, 1094)\n",
      "[M::mem_pestat] mean and std.dev: (877.65, 63.30)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (572, 1181)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (341, 1508, 1508)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3842)\n",
      "[M::mem_pestat] mean and std.dev: (809.67, 636.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5009)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 104196 reads in 6.191 CPU sec, 6.235 real sec\n",
      "[M::process] read 102798 sequences (10000098 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 48932, 4, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 871, 920)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (659, 1094)\n",
      "[M::mem_pestat] mean and std.dev: (877.19, 63.63)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (572, 1181)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 103964 reads in 6.417 CPU sec, 6.622 real sec\n",
      "[M::process] read 103872 sequences (10000144 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48534, 6, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 872, 921)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (657, 1097)\n",
      "[M::mem_pestat] mean and std.dev: (877.88, 63.76)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (569, 1185)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 102798 reads in 6.098 CPU sec, 6.117 real sec\n",
      "[M::process] read 31674 sequences (3075333 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 48988, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 871, 920)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (659, 1094)\n",
      "[M::mem_pestat] mean and std.dev: (877.25, 63.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (572, 1181)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 103872 reads in 6.710 CPU sec, 8.412 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 14936, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (834, 871, 920)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (662, 1092)\n",
      "[M::mem_pestat] mean and std.dev: (877.24, 63.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (576, 1178)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 31674 reads in 2.569 CPU sec, 2.602 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/assembly.fasta /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492188_1.fastq /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492188_2.fastq\n",
      "[main] Real time: 149.916 sec; CPU: 128.103 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 118130 sequences (10000021 bp)...\n",
      "[M::process] read 118102 sequences (10000080 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 55180, 3, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (832, 869, 918)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (660, 1090)\n",
      "[M::mem_pestat] mean and std.dev: (875.92, 63.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (574, 1176)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 118130 reads in 6.411 CPU sec, 7.889 real sec\n",
      "[M::process] read 117778 sequences (10000067 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54908, 5, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (833, 872, 921)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (657, 1097)\n",
      "[M::mem_pestat] mean and std.dev: (877.56, 64.05)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (569, 1185)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 118102 reads in 6.579 CPU sec, 8.193 real sec\n",
      "[M::process] read 117250 sequences (10000092 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54999, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (834, 872, 922)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (658, 1098)\n",
      "[M::mem_pestat] mean and std.dev: (878.70, 63.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (570, 1186)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 117778 reads in 6.206 CPU sec, 6.599 real sec\n",
      "[M::process] read 116972 sequences (10000036 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54739, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (834, 872, 922)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (658, 1098)\n",
      "[M::mem_pestat] mean and std.dev: (878.36, 64.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (570, 1186)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 117250 reads in 6.267 CPU sec, 6.962 real sec\n",
      "[M::process] read 116702 sequences (10000060 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 54656, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (834, 872, 922)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (658, 1098)\n",
      "[M::mem_pestat] mean and std.dev: (878.77, 63.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (570, 1186)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 116972 reads in 6.227 CPU sec, 6.673 real sec\n",
      "[M::process] read 116726 sequences (10000042 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54680, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (834, 872, 922)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (658, 1098)\n",
      "[M::mem_pestat] mean and std.dev: (878.58, 63.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (570, 1186)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 116702 reads in 5.904 CPU sec, 5.901 real sec\n",
      "[M::process] read 116238 sequences (10000068 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54711, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (834, 873, 922)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (658, 1098)\n",
      "[M::mem_pestat] mean and std.dev: (878.87, 63.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (570, 1186)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 116726 reads in 6.162 CPU sec, 6.745 real sec\n",
      "[M::process] read 116046 sequences (10000009 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (6, 54638, 2, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (835, 873, 922)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (661, 1096)\n",
      "[M::mem_pestat] mean and std.dev: (878.98, 63.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (574, 1183)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 116238 reads in 5.887 CPU sec, 5.893 real sec\n",
      "[M::process] read 115298 sequences (10000047 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54470, 4, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (835, 874, 923)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (659, 1099)\n",
      "[M::mem_pestat] mean and std.dev: (880.00, 63.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (571, 1187)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 116046 reads in 5.932 CPU sec, 6.021 real sec\n",
      "[M::process] read 114206 sequences (10000139 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54313, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (835, 874, 924)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (657, 1102)\n",
      "[M::mem_pestat] mean and std.dev: (879.83, 64.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (568, 1191)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 115298 reads in 5.901 CPU sec, 5.908 real sec\n",
      "[M::process] read 114808 sequences (10000138 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 54006, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (834, 873, 923)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (656, 1101)\n",
      "[M::mem_pestat] mean and std.dev: (879.53, 64.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (567, 1190)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114206 reads in 5.736 CPU sec, 5.709 real sec\n",
      "[M::process] read 113812 sequences (10000173 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54311, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (835, 873, 923)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (659, 1099)\n",
      "[M::mem_pestat] mean and std.dev: (879.68, 64.10)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (571, 1187)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114808 reads in 5.731 CPU sec, 5.764 real sec\n",
      "[M::process] read 113656 sequences (10000011 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53993, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (834, 873, 923)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (656, 1101)\n",
      "[M::mem_pestat] mean and std.dev: (879.33, 63.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (567, 1190)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 113812 reads in 5.678 CPU sec, 5.715 real sec\n",
      "[M::process] read 113564 sequences (10000081 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53993, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (835, 873, 923)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (659, 1099)\n",
      "[M::mem_pestat] mean and std.dev: (879.34, 63.87)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (571, 1187)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 113656 reads in 5.628 CPU sec, 5.659 real sec\n",
      "[M::process] read 113572 sequences (10000097 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 54013, 6, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (835, 873, 922)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (661, 1096)\n",
      "[M::mem_pestat] mean and std.dev: (879.55, 63.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (574, 1183)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 113564 reads in 5.920 CPU sec, 6.516 real sec\n",
      "[M::process] read 113182 sequences (10000198 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53910, 8, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (834, 874, 924)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (654, 1104)\n",
      "[M::mem_pestat] mean and std.dev: (879.51, 64.60)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (564, 1194)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 113572 reads in 6.064 CPU sec, 6.841 real sec\n",
      "[M::process] read 33958 sequences (2982893 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 53673, 4, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (834, 873, 922)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (658, 1098)\n",
      "[M::mem_pestat] mean and std.dev: (879.15, 63.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (570, 1186)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 113182 reads in 6.622 CPU sec, 6.941 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 16064, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (835, 873, 921)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (663, 1093)\n",
      "[M::mem_pestat] mean and std.dev: (879.13, 62.70)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (577, 1179)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 33958 reads in 1.983 CPU sec, 1.892 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/assembly.fasta /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492189_1.fastq /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492189_2.fastq\n",
      "[main] Real time: 112.106 sec; CPU: 105.074 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 102954 sequences (10000046 bp)...\n",
      "[M::process] read 102486 sequences (10000065 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47058, 5, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (760, 792, 830)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (620, 970)\n",
      "[M::mem_pestat] mean and std.dev: (794.67, 51.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (550, 1040)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 102954 reads in 8.725 CPU sec, 9.540 real sec\n",
      "[M::process] read 102790 sequences (10000159 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47358, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (760, 793, 830)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (620, 970)\n",
      "[M::mem_pestat] mean and std.dev: (795.39, 50.80)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (550, 1040)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 102486 reads in 7.910 CPU sec, 8.369 real sec\n",
      "[M::process] read 102524 sequences (10000008 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 46526, 8, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (760, 792, 829)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (622, 967)\n",
      "[M::mem_pestat] mean and std.dev: (794.28, 51.21)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (553, 1036)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 102790 reads in 9.406 CPU sec, 10.297 real sec\n",
      "[M::process] read 103230 sequences (10000060 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47013, 9, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (761, 792, 830)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (623, 968)\n",
      "[M::mem_pestat] mean and std.dev: (795.29, 50.75)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (554, 1037)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 102524 reads in 8.815 CPU sec, 9.718 real sec\n",
      "[M::process] read 102948 sequences (10000091 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 46893, 9, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (760, 792, 829)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (622, 967)\n",
      "[M::mem_pestat] mean and std.dev: (794.55, 50.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (553, 1036)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 103230 reads in 9.173 CPU sec, 9.876 real sec\n",
      "[M::process] read 102986 sequences (10000043 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 45041, 5, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (759, 791, 829)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (619, 969)\n",
      "[M::mem_pestat] mean and std.dev: (793.50, 51.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (549, 1039)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 102948 reads in 9.484 CPU sec, 10.055 real sec\n",
      "[M::process] read 102986 sequences (10000169 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 44520, 13, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (759, 791, 828)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (621, 966)\n",
      "[M::mem_pestat] mean and std.dev: (793.33, 51.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (552, 1035)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (75, 140, 597)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1641)\n",
      "[M::mem_pestat] mean and std.dev: (312.54, 321.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2163)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 102986 reads in 9.572 CPU sec, 10.003 real sec\n",
      "[M::process] read 103306 sequences (10000173 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 42665, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (760, 792, 829)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (622, 967)\n",
      "[M::mem_pestat] mean and std.dev: (794.30, 50.86)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (553, 1036)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 102986 reads in 9.343 CPU sec, 9.710 real sec\n",
      "[M::process] read 103266 sequences (10000021 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 42869, 12, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (759, 791, 829)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (619, 969)\n",
      "[M::mem_pestat] mean and std.dev: (793.75, 51.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (549, 1039)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (68, 161, 587)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1625)\n",
      "[M::mem_pestat] mean and std.dev: (343.17, 345.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2144)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 103306 reads in 10.127 CPU sec, 11.299 real sec\n",
      "[M::process] read 102786 sequences (10000199 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 45786, 11, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (758, 791, 828)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (618, 968)\n",
      "[M::mem_pestat] mean and std.dev: (793.41, 51.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (548, 1038)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (158, 172, 552)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1340)\n",
      "[M::mem_pestat] mean and std.dev: (322.64, 313.39)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1734)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 103266 reads in 8.718 CPU sec, 9.410 real sec\n",
      "[M::process] read 102198 sequences (10000117 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47686, 7, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (760, 793, 831)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (618, 973)\n",
      "[M::mem_pestat] mean and std.dev: (795.20, 51.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (547, 1044)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 102786 reads in 7.446 CPU sec, 8.510 real sec\n",
      "[M::process] read 101956 sequences (10000117 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47533, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (761, 793, 830)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (623, 968)\n",
      "[M::mem_pestat] mean and std.dev: (795.41, 50.26)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (554, 1037)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 102198 reads in 6.695 CPU sec, 7.066 real sec\n",
      "[M::process] read 101794 sequences (10000189 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47749, 10, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (761, 793, 831)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (621, 971)\n",
      "[M::mem_pestat] mean and std.dev: (795.84, 50.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (551, 1041)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (102, 227, 732)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1992)\n",
      "[M::mem_pestat] mean and std.dev: (356.90, 318.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 2622)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 101956 reads in 6.675 CPU sec, 7.527 real sec\n",
      "[M::process] read 101924 sequences (10000197 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47547, 9, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (761, 793, 831)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (621, 971)\n",
      "[M::mem_pestat] mean and std.dev: (795.75, 50.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (551, 1041)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 101794 reads in 6.992 CPU sec, 8.006 real sec\n",
      "[M::process] read 101934 sequences (10000011 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47173, 7, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (760, 792, 829)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (622, 967)\n",
      "[M::mem_pestat] mean and std.dev: (794.62, 50.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (553, 1036)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 101924 reads in 7.978 CPU sec, 8.248 real sec\n",
      "[M::process] read 101676 sequences (10000170 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47144, 13, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (760, 793, 831)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (618, 973)\n",
      "[M::mem_pestat] mean and std.dev: (795.24, 51.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (547, 1044)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (128, 149, 423)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1013)\n",
      "[M::mem_pestat] mean and std.dev: (250.15, 217.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1308)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 101934 reads in 8.193 CPU sec, 8.545 real sec\n",
      "[M::process] read 101646 sequences (10000003 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47418, 14, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (760, 792, 830)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (620, 970)\n",
      "[M::mem_pestat] mean and std.dev: (794.86, 51.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (550, 1040)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (45, 47, 510)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 1440)\n",
      "[M::mem_pestat] mean and std.dev: (198.85, 229.49)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 1905)\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_process_seqs] Processed 101676 reads in 6.532 CPU sec, 7.232 real sec\n",
      "[M::process] read 101602 sequences (10000061 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47479, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (760, 792, 830)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (620, 970)\n",
      "[M::mem_pestat] mean and std.dev: (794.83, 50.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (550, 1040)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 101646 reads in 6.693 CPU sec, 7.186 real sec\n",
      "[M::process] read 53008 sequences (5217533 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 47541, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (760, 792, 830)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (620, 970)\n",
      "[M::mem_pestat] mean and std.dev: (795.17, 50.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (550, 1040)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 101602 reads in 7.225 CPU sec, 8.678 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 24730, 6, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (761, 792, 831)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (621, 971)\n",
      "[M::mem_pestat] mean and std.dev: (795.24, 50.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (551, 1041)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 53008 reads in 4.086 CPU sec, 4.180 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/assembly.fasta /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492190_1.fastq /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492190_2.fastq\n",
      "[main] Real time: 173.827 sec; CPU: 160.058 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 118884 sequences (10000024 bp)...\n",
      "[M::process] read 118632 sequences (10000173 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54403, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (773, 905, 959)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (401, 1331)\n",
      "[M::mem_pestat] mean and std.dev: (861.00, 150.83)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (215, 1517)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 118884 reads in 6.149 CPU sec, 6.414 real sec\n",
      "[M::process] read 117972 sequences (10000146 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 54344, 7, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (787, 906, 959)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (443, 1303)\n",
      "[M::mem_pestat] mean and std.dev: (869.62, 140.88)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (271, 1475)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 118632 reads in 6.155 CPU sec, 6.139 real sec\n",
      "[M::process] read 117558 sequences (10000124 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 53978, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (798, 908, 961)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (472, 1287)\n",
      "[M::mem_pestat] mean and std.dev: (876.36, 134.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (309, 1450)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 117972 reads in 6.908 CPU sec, 7.806 real sec\n",
      "[M::process] read 117650 sequences (10000011 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53967, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (796, 908, 961)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (466, 1291)\n",
      "[M::mem_pestat] mean and std.dev: (875.15, 136.00)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (301, 1456)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 117558 reads in 6.659 CPU sec, 7.901 real sec\n",
      "[M::process] read 117194 sequences (10000007 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 54002, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (804, 909, 961)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (490, 1275)\n",
      "[M::mem_pestat] mean and std.dev: (879.70, 129.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (333, 1432)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 117650 reads in 6.735 CPU sec, 8.515 real sec\n",
      "[M::process] read 115686 sequences (10000092 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 54065, 7, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (806, 910, 962)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (494, 1274)\n",
      "[M::mem_pestat] mean and std.dev: (881.25, 128.95)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (338, 1430)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 117194 reads in 6.465 CPU sec, 6.887 real sec\n",
      "[M::process] read 115494 sequences (10000096 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53744, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (811, 910, 963)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (507, 1267)\n",
      "[M::mem_pestat] mean and std.dev: (884.44, 126.03)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (355, 1419)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 115686 reads in 6.070 CPU sec, 6.046 real sec\n",
      "[M::process] read 114936 sequences (10000035 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53612, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (809, 909, 963)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (501, 1271)\n",
      "[M::mem_pestat] mean and std.dev: (883.26, 126.93)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (347, 1425)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 115494 reads in 6.040 CPU sec, 6.065 real sec\n",
      "[M::process] read 114862 sequences (10000129 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 53489, 6, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (811, 911, 963)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (507, 1267)\n",
      "[M::mem_pestat] mean and std.dev: (884.82, 126.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (355, 1419)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114936 reads in 6.704 CPU sec, 7.813 real sec\n",
      "[M::process] read 114576 sequences (10000151 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53523, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (809, 909, 963)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (501, 1271)\n",
      "[M::mem_pestat] mean and std.dev: (883.58, 127.08)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (347, 1425)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114862 reads in 6.348 CPU sec, 6.942 real sec\n",
      "[M::process] read 114756 sequences (10000170 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53446, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (806, 910, 964)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (490, 1280)\n",
      "[M::mem_pestat] mean and std.dev: (881.32, 130.74)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (332, 1438)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114576 reads in 6.215 CPU sec, 6.498 real sec\n",
      "[M::process] read 114966 sequences (10000054 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53382, 5, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (805, 909, 962)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (491, 1276)\n",
      "[M::mem_pestat] mean and std.dev: (881.23, 129.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (334, 1433)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114756 reads in 6.197 CPU sec, 6.584 real sec\n",
      "[M::process] read 114850 sequences (10000163 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 53463, 2, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (800, 909, 962)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (476, 1286)\n",
      "[M::mem_pestat] mean and std.dev: (877.86, 133.78)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (314, 1448)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114966 reads in 6.453 CPU sec, 7.421 real sec\n",
      "[M::process] read 114702 sequences (10000000 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 53533, 8, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (802, 910, 963)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (480, 1285)\n",
      "[M::mem_pestat] mean and std.dev: (879.43, 132.09)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (319, 1446)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114850 reads in 6.760 CPU sec, 8.325 real sec\n",
      "[M::process] read 114394 sequences (10000128 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53536, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (803, 909, 963)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (483, 1283)\n",
      "[M::mem_pestat] mean and std.dev: (879.88, 131.81)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (323, 1443)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114702 reads in 6.500 CPU sec, 6.829 real sec\n",
      "[M::process] read 114290 sequences (10000151 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53296, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (803, 909, 963)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (483, 1283)\n",
      "[M::mem_pestat] mean and std.dev: (879.86, 131.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (323, 1443)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114394 reads in 6.483 CPU sec, 6.838 real sec\n",
      "[M::process] read 28598 sequences (2522056 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53259, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (811, 909, 962)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (509, 1264)\n",
      "[M::mem_pestat] mean and std.dev: (884.74, 125.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (358, 1415)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 114290 reads in 6.992 CPU sec, 7.386 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 13322, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (800, 910, 962)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (476, 1286)\n",
      "[M::mem_pestat] mean and std.dev: (878.93, 133.79)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (314, 1448)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 28598 reads in 1.848 CPU sec, 1.761 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/assembly.fasta /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492193_1.fastq /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492193_2.fastq\n",
      "[main] Real time: 122.439 sec; CPU: 111.900 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 104930 sequences (10000111 bp)...\n",
      "[M::process] read 104974 sequences (10000183 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47077, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (839, 875, 922)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (673, 1088)\n",
      "[M::mem_pestat] mean and std.dev: (881.30, 60.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (590, 1171)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104930 reads in 9.559 CPU sec, 10.260 real sec\n",
      "[M::process] read 105146 sequences (10000013 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47893, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (840, 876, 922)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (676, 1086)\n",
      "[M::mem_pestat] mean and std.dev: (882.52, 59.90)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (594, 1168)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104974 reads in 7.806 CPU sec, 8.152 real sec\n",
      "[M::process] read 104860 sequences (10000185 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47619, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (840, 876, 923)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (674, 1089)\n",
      "[M::mem_pestat] mean and std.dev: (882.40, 60.25)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (591, 1172)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 105146 reads in 8.377 CPU sec, 8.838 real sec\n",
      "[M::process] read 104600 sequences (10000059 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47618, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (840, 877, 923)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (674, 1089)\n",
      "[M::mem_pestat] mean and std.dev: (882.73, 60.16)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (591, 1172)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104860 reads in 9.314 CPU sec, 9.576 real sec\n",
      "[M::process] read 104892 sequences (10000063 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47149, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (839, 876, 923)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (671, 1091)\n",
      "[M::mem_pestat] mean and std.dev: (882.22, 60.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (587, 1175)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104600 reads in 7.948 CPU sec, 8.161 real sec\n",
      "[M::process] read 104428 sequences (10000102 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47935, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (840, 877, 924)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (672, 1092)\n",
      "[M::mem_pestat] mean and std.dev: (882.87, 60.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (588, 1176)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104892 reads in 7.780 CPU sec, 8.089 real sec\n",
      "[M::process] read 105634 sequences (10000132 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47513, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (841, 877, 923)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (677, 1087)\n",
      "[M::mem_pestat] mean and std.dev: (883.00, 60.04)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (595, 1169)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104428 reads in 8.258 CPU sec, 8.664 real sec\n",
      "[M::process] read 105092 sequences (10000038 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47304, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (840, 877, 923)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (674, 1089)\n",
      "[M::mem_pestat] mean and std.dev: (882.25, 60.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (591, 1172)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 105634 reads in 8.270 CPU sec, 8.560 real sec\n",
      "[M::process] read 105084 sequences (10000014 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 45119, 5, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (839, 876, 921)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (675, 1085)\n",
      "[M::mem_pestat] mean and std.dev: (881.66, 59.98)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (593, 1167)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 105092 reads in 8.849 CPU sec, 9.102 real sec\n",
      "[M::process] read 105280 sequences (10000132 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47514, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (841, 877, 924)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (675, 1090)\n",
      "[M::mem_pestat] mean and std.dev: (883.38, 59.82)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (592, 1173)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 105084 reads in 8.629 CPU sec, 8.797 real sec\n",
      "[M::process] read 105492 sequences (10000037 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 46634, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (840, 876, 923)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (674, 1089)\n",
      "[M::mem_pestat] mean and std.dev: (882.25, 59.91)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (591, 1172)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 105280 reads in 9.657 CPU sec, 10.151 real sec\n",
      "[M::process] read 104764 sequences (10000076 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 46619, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (839, 875, 921)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (675, 1085)\n",
      "[M::mem_pestat] mean and std.dev: (881.24, 59.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (593, 1167)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 105492 reads in 9.133 CPU sec, 9.473 real sec\n",
      "[M::process] read 104992 sequences (10000090 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 47290, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (840, 877, 925)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (670, 1095)\n",
      "[M::mem_pestat] mean and std.dev: (883.46, 60.94)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (585, 1180)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104764 reads in 9.184 CPU sec, 9.500 real sec\n",
      "[M::process] read 104646 sequences (10000054 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 46759, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (840, 877, 923)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (674, 1089)\n",
      "[M::mem_pestat] mean and std.dev: (882.60, 60.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (591, 1172)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104992 reads in 8.590 CPU sec, 8.994 real sec\n",
      "[M::process] read 106284 sequences (10000012 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47445, 2, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (841, 877, 925)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (673, 1093)\n",
      "[M::mem_pestat] mean and std.dev: (883.21, 60.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (589, 1177)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104646 reads in 8.003 CPU sec, 8.264 real sec\n",
      "[M::process] read 105456 sequences (10000138 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47873, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (840, 877, 924)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (672, 1092)\n",
      "[M::mem_pestat] mean and std.dev: (882.85, 60.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (588, 1176)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 106284 reads in 8.595 CPU sec, 8.737 real sec\n",
      "[M::process] read 105074 sequences (10000162 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 46014, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (839, 875, 922)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (673, 1088)\n",
      "[M::mem_pestat] mean and std.dev: (881.47, 60.14)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (590, 1171)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 105456 reads in 8.641 CPU sec, 8.860 real sec\n",
      "[M::process] read 104680 sequences (10000190 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 45833, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (839, 875, 921)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (675, 1085)\n",
      "[M::mem_pestat] mean and std.dev: (880.72, 59.68)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (593, 1167)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 105074 reads in 9.378 CPU sec, 9.579 real sec\n",
      "[M::process] read 3692 sequences (337450 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 47118, 3, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (839, 875, 921)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (675, 1085)\n",
      "[M::mem_pestat] mean and std.dev: (881.25, 60.27)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (593, 1167)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 104680 reads in 11.947 CPU sec, 12.471 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 1594, 0, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (840, 877, 924)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (672, 1092)\n",
      "[M::mem_pestat] mean and std.dev: (883.42, 59.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (588, 1176)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 3692 reads in 0.567 CPU sec, 0.376 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/assembly.fasta /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492194_1.fastq /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492194_2.fastq\n",
      "[main] Real time: 174.808 sec; CPU: 168.652 sec\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 117534 sequences (10000142 bp)...\n",
      "[M::process] read 112626 sequences (10000072 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 55764, 8, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (342, 378, 423)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (180, 585)\n",
      "[M::mem_pestat] mean and std.dev: (384.44, 51.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (99, 666)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 117534 reads in 6.029 CPU sec, 6.469 real sec\n",
      "[M::process] read 110202 sequences (10000099 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53891, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (345, 382, 428)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (179, 594)\n",
      "[M::mem_pestat] mean and std.dev: (387.89, 52.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (96, 677)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 112626 reads in 5.677 CPU sec, 5.684 real sec\n",
      "[M::process] read 110020 sequences (10000001 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 52965, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 386, 431)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (179, 599)\n",
      "[M::mem_pestat] mean and std.dev: (390.18, 52.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (95, 683)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 110202 reads in 5.617 CPU sec, 5.676 real sec\n",
      "[M::process] read 110412 sequences (10000106 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (4, 52897, 1, 2)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 386, 432)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 602)\n",
      "[M::mem_pestat] mean and std.dev: (390.48, 52.69)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (92, 687)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 110020 reads in 5.704 CPU sec, 5.816 real sec\n",
      "[M::process] read 110128 sequences (10000023 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 53062, 2, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 387, 432)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 602)\n",
      "[M::mem_pestat] mean and std.dev: (390.72, 52.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (92, 687)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 110412 reads in 5.652 CPU sec, 5.725 real sec\n",
      "[M::process] read 109684 sequences (10000082 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 53028, 2, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (348, 387, 433)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (178, 603)\n",
      "[M::mem_pestat] mean and std.dev: (391.22, 52.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (93, 688)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 110128 reads in 5.698 CPU sec, 5.970 real sec\n",
      "[M::process] read 109172 sequences (10000135 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 52865, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 387, 433)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (175, 605)\n",
      "[M::mem_pestat] mean and std.dev: (390.85, 52.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (89, 691)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 109684 reads in 5.497 CPU sec, 5.503 real sec\n",
      "[M::process] read 109040 sequences (10000052 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 52583, 6, 5)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 386, 432)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 602)\n",
      "[M::mem_pestat] mean and std.dev: (390.67, 52.73)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (92, 687)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 109172 reads in 5.691 CPU sec, 5.994 real sec\n",
      "[M::process] read 108742 sequences (10000092 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2, 52501, 2, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 387, 432)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 602)\n",
      "[M::mem_pestat] mean and std.dev: (390.87, 52.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (92, 687)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 109040 reads in 5.656 CPU sec, 5.756 real sec\n",
      "[M::process] read 108556 sequences (10000029 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 52501, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 387, 432)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 602)\n",
      "[M::mem_pestat] mean and std.dev: (390.74, 52.51)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (92, 687)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 108742 reads in 5.679 CPU sec, 5.770 real sec\n",
      "[M::process] read 108730 sequences (10000137 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 52461, 2, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 387, 432)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 602)\n",
      "[M::mem_pestat] mean and std.dev: (390.68, 52.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (92, 687)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 108556 reads in 5.666 CPU sec, 5.700 real sec\n",
      "[M::process] read 108498 sequences (10000100 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 52487, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 386, 431)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (179, 599)\n",
      "[M::mem_pestat] mean and std.dev: (390.17, 52.52)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (95, 683)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 108730 reads in 5.662 CPU sec, 5.705 real sec\n",
      "[M::process] read 108372 sequences (10000010 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 52390, 1, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 387, 433)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (175, 605)\n",
      "[M::mem_pestat] mean and std.dev: (391.03, 52.58)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (89, 691)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 108498 reads in 5.662 CPU sec, 5.704 real sec\n",
      "[M::process] read 108004 sequences (10000095 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 52320, 4, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 387, 432)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 602)\n",
      "[M::mem_pestat] mean and std.dev: (390.87, 52.72)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (92, 687)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 108372 reads in 5.688 CPU sec, 5.793 real sec\n",
      "[M::process] read 107786 sequences (10000087 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (0, 52125, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (348, 387, 433)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (178, 603)\n",
      "[M::mem_pestat] mean and std.dev: (391.15, 52.65)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (93, 688)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 108004 reads in 5.763 CPU sec, 5.858 real sec\n",
      "[M::process] read 107830 sequences (10000024 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 52092, 6, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 387, 432)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 602)\n",
      "[M::mem_pestat] mean and std.dev: (390.67, 52.48)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (92, 687)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 107786 reads in 5.596 CPU sec, 5.716 real sec\n",
      "[M::process] read 107692 sequences (10000111 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 52151, 4, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 387, 432)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 602)\n",
      "[M::mem_pestat] mean and std.dev: (390.89, 52.42)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (92, 687)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 107830 reads in 5.628 CPU sec, 5.707 real sec\n",
      "[M::process] read 26972 sequences (2510659 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3, 52076, 3, 0)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (347, 387, 432)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (177, 602)\n",
      "[M::mem_pestat] mean and std.dev: (390.80, 52.36)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (92, 687)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 107692 reads in 6.013 CPU sec, 6.135 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (1, 13082, 1, 1)\n",
      "[M::mem_pestat] skip orientation FF as there are not enough pairs\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (346, 386, 431)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (176, 601)\n",
      "[M::mem_pestat] mean and std.dev: (389.87, 52.13)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (91, 686)\n",
      "[M::mem_pestat] skip orientation RF as there are not enough pairs\n",
      "[M::mem_pestat] skip orientation RR as there are not enough pairs\n",
      "[M::mem_process_seqs] Processed 26972 reads in 1.616 CPU sec, 1.499 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/assembly.fasta /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492197_1.fastq /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR492197_2.fastq\n",
      "[main] Real time: 106.421 sec; CPU: 104.406 sec\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for SAMPLE in 83 86 88 89 90 93 94 97\n",
    "do\n",
    "bwa mem /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/assembly.fasta /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR4921${SAMPLE}_1.fastq /Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/READS/SRR4921${SAMPLE}_2.fastq > ${SAMPLE}.sam\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 2: syntax error near unexpected token `samtools'\n",
      "bash: line 2: `samtools view -S -b ${SAMPLE}.sam | samtools sort -o ${SAMPLE}.sorted.bam'\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "for SAMPLE in 83 86 88 89 90 93 94 97\n",
    "do\n",
    "samtools view -S -b ../${SAMPLE}.sam | samtools sort -o ../${SAMPLE}.sorted.bam\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#create a metabat2 enviornment\n",
    "#conda create -n metabat2 -c bioconda metabat2=2.13 libdeflate=1.0"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#activate metabat2 envionment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#run in command line\n",
    "#jgi_summarize_bam_contig_depths --outputDepth depth.txt *.sorted.bam\n",
    "\n",
    "#conda activate metabat\n",
    "#metabat2 -i week13_data/assembly.fasta -a depth.txt -o week13_data/mybins"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Answers to questions - 1 --> 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "# step 1 question \n",
    "# Enterococcus faecalis was generally more abundant than other species thoughout all days although it did vary from 90-60% abundance. The First day has a relatively high abundance of Staphylococcus which quickly reduces to a low basal amount.\n",
    "\n",
    "# step 2 question 2 \n",
    "# kamer counting\n",
    "\n",
    "# Question 3\n",
    "# a) 6 binzzzzzz\n",
    "#b) ~ 33%\n",
    "#c) since each bin were about 1-3 mb, this would seem to be about right.\n",
    "#d) Use a known reference genome to comare how closely the bin allignes to the reference genome in order to see bin completeness. Use missalignments to dertemine contamination. \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# part 3"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "f = open(\"/Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/week13_data/KRAKEN/assembly.kraken\")\n",
    "assembly=f.readlines()\n",
    "f.close()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# junk broken code\n",
    "\n",
    "def bin_species_id(bin_num, assembly):\n",
    "    f= open(\"/Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/bins/bin.\"+bin_num+\".fa\")\n",
    "    binfile=f.readlines()\n",
    "    f.close()\n",
    "    \n",
    "    Dict_meow ={}\n",
    "    for line in binfile:\n",
    "        if '>' in line:\n",
    "            node = str(line.rstrip('\\t')[1:])\n",
    "            #print(node)\n",
    "            for line in assembly:\n",
    "                if node in line:\n",
    "                    key =line.split('\\t')[1]\n",
    "                    \n",
    "                    #print(line.split('\\t'))\n",
    "                    Dict_meow.setdefault(key,0)\n",
    "                    Dict_meow[key] += 1\n",
    "    print(Dict_meow)\n",
    "        \n",
    "        \n",
    "    f=open(\"/Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/bins/bin_species.\"+bin_num+\".fa\",'w')\n",
    "    for key,val in Dict_meow.items():\n",
    "        f.write(key+ '\\n')\n",
    "        f.write(str(val)+'\\n')\n",
    "        f.write('\\n')\n",
    "        \n",
    "    max_key = max(Dict_meow, key=lambda k: Dict_meow[k])\n",
    "    f.write('\\nMost associated species: \\t'+max_key)\n",
    "    \n",
    "    f.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "def bin_species_id(bin_num, assembly):\n",
    "    f = open(\"/Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/bins/bin.\"+bin_num+\".fa\")\n",
    "    binfile = f.readlines()\n",
    "    f.close()\n",
    "    bin_dict = {}\n",
    "    for line in binfile:\n",
    "        if '>' in line:\n",
    "            node = str(line.strip()[1:])\n",
    "            for line in assembly:\n",
    "                if node in line:\n",
    "                    key = line.split('\\t')[1]\n",
    "                    bin_dict.setdefault(key,0)\n",
    "                    bin_dict[key] += 1\n",
    "                    \n",
    "    f = open(\"/Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/bins/bin_species.\"+bin_num+\".fa\",'w')\n",
    "    for key,val in bin_dict.items():\n",
    "        f.write(key + '\\n')\n",
    "        f.write(str(val)+'\\n')\n",
    "        f.write('\\n')\n",
    "    max_key = max(bin_dict, key=lambda k: bin_dict[k])\n",
    "    f.write('\\nMost likely associated species:\\t'+max_key)\n",
    "    f.close()\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "\n",
    "bin_nums = ['1','2','3','4','5','6','7','8']\n",
    "for num in bin_nums:\n",
    "    bin_species_id(num,assembly)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Question 4 \n",
    "## Nameing bins (a)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "bin.1 Staphylococcus haemolyticus JCSC1435\n",
    "\n",
    "bin.2 Leuconostoc citreum KM20\n",
    "\n",
    "bin 3 Staphylococcus lugdunensis HKU09-01\n",
    "\n",
    "bin 4 Enterococcus faecalis OG1RF\n",
    "\n",
    "bin 5 Cutibacterium avidum 44067\n",
    "\n",
    "bin 6 Staphylococcus epidermidis RP62A\n",
    "\n",
    "bin 7 Staphylococcus aureus subsp. aureus CN1\n",
    "\n",
    "bin 8 Anaerococcus prevotii DSM 20548"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## quantitative approach (b)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Use 16s ribosomal rRNA conserved sequences to quantify the number of individual species types and infer higher taxa quantities. This will not work for vial species."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "df=pd.read_csv(\"/Users/cmdb/qbb2020-answers/QUANT2020/data/Lab10_Metagenomics/abundance_table.tab\", delimiter='\\t',index_col='Genomic bins')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "import seaborn as sb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "df.rename(index = {'bin.1':'Staphylococcus haemolyticus', 'bin.4':'Enterococcus faecalis', 'bin.2':'Leuconostoc citreum', 'bin.5':'Cutibacterium avidum','bin.6':'Staphylococcus epidermidis', 'bin.7':'Staphylococcus aureus','bin.8':'Anaerococcus prevotii', 'bin.3':'Staphylococcus lugdunensis'}, inplace = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/cmdb/miniconda3/envs/cmdb/lib/python3.6/site-packages/pandas/core/frame.py:4301: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>day1</th>\n",
       "      <th>day2</th>\n",
       "      <th>day3</th>\n",
       "      <th>day4</th>\n",
       "      <th>day5</th>\n",
       "      <th>day6</th>\n",
       "      <th>day7</th>\n",
       "      <th>day8</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Genomic bins</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>Staphylococcus haemolyticus</th>\n",
       "      <td>50.103235</td>\n",
       "      <td>167.273691</td>\n",
       "      <td>102.432386</td>\n",
       "      <td>97.521190</td>\n",
       "      <td>14.523105</td>\n",
       "      <td>172.449773</td>\n",
       "      <td>53.357374</td>\n",
       "      <td>72.544371</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Enterococcus faecalis</th>\n",
       "      <td>4115.242526</td>\n",
       "      <td>9348.741359</td>\n",
       "      <td>8139.870719</td>\n",
       "      <td>8308.156703</td>\n",
       "      <td>5267.858711</td>\n",
       "      <td>6267.155205</td>\n",
       "      <td>6550.505763</td>\n",
       "      <td>8745.103316</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Leuconostoc citreum</th>\n",
       "      <td>1.751119</td>\n",
       "      <td>8.887730</td>\n",
       "      <td>5.838109</td>\n",
       "      <td>5.929623</td>\n",
       "      <td>4.583909</td>\n",
       "      <td>1.804228</td>\n",
       "      <td>3.259988</td>\n",
       "      <td>19.275630</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Cutibacterium avidum</th>\n",
       "      <td>1566.783656</td>\n",
       "      <td>0.552469</td>\n",
       "      <td>0.875860</td>\n",
       "      <td>0.356678</td>\n",
       "      <td>1.251430</td>\n",
       "      <td>315.740755</td>\n",
       "      <td>457.543378</td>\n",
       "      <td>4250.653466</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus epidermidis</th>\n",
       "      <td>1335.147536</td>\n",
       "      <td>820.872805</td>\n",
       "      <td>1033.761424</td>\n",
       "      <td>999.273162</td>\n",
       "      <td>1071.641137</td>\n",
       "      <td>1670.022745</td>\n",
       "      <td>1187.660153</td>\n",
       "      <td>936.681889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus aureus</th>\n",
       "      <td>1.287979</td>\n",
       "      <td>1.452063</td>\n",
       "      <td>1.347894</td>\n",
       "      <td>0.747525</td>\n",
       "      <td>3.287205</td>\n",
       "      <td>6.459688</td>\n",
       "      <td>1124.427189</td>\n",
       "      <td>590.194971</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Anaerococcus prevotii</th>\n",
       "      <td>0.186461</td>\n",
       "      <td>0.065351</td>\n",
       "      <td>4.223711</td>\n",
       "      <td>0.979570</td>\n",
       "      <td>0.259990</td>\n",
       "      <td>1.211367</td>\n",
       "      <td>73.851083</td>\n",
       "      <td>3366.910935</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Staphylococcus lugdunensis</th>\n",
       "      <td>2.555610</td>\n",
       "      <td>6.194380</td>\n",
       "      <td>0.475050</td>\n",
       "      <td>0.333899</td>\n",
       "      <td>0.491073</td>\n",
       "      <td>78.291586</td>\n",
       "      <td>6.639770</td>\n",
       "      <td>10.595311</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    day1         day2         day3  \\\n",
       "Genomic bins                                                         \n",
       "Staphylococcus haemolyticus    50.103235   167.273691   102.432386   \n",
       "Enterococcus faecalis        4115.242526  9348.741359  8139.870719   \n",
       "Leuconostoc citreum             1.751119     8.887730     5.838109   \n",
       "Cutibacterium avidum         1566.783656     0.552469     0.875860   \n",
       "Staphylococcus epidermidis   1335.147536   820.872805  1033.761424   \n",
       "Staphylococcus aureus           1.287979     1.452063     1.347894   \n",
       "Anaerococcus prevotii           0.186461     0.065351     4.223711   \n",
       "Staphylococcus lugdunensis      2.555610     6.194380     0.475050   \n",
       "\n",
       "                                    day4         day5         day6  \\\n",
       "Genomic bins                                                         \n",
       "Staphylococcus haemolyticus    97.521190    14.523105   172.449773   \n",
       "Enterococcus faecalis        8308.156703  5267.858711  6267.155205   \n",
       "Leuconostoc citreum             5.929623     4.583909     1.804228   \n",
       "Cutibacterium avidum            0.356678     1.251430   315.740755   \n",
       "Staphylococcus epidermidis    999.273162  1071.641137  1670.022745   \n",
       "Staphylococcus aureus           0.747525     3.287205     6.459688   \n",
       "Anaerococcus prevotii           0.979570     0.259990     1.211367   \n",
       "Staphylococcus lugdunensis      0.333899     0.491073    78.291586   \n",
       "\n",
       "                                    day7         day8  \n",
       "Genomic bins                                           \n",
       "Staphylococcus haemolyticus    53.357374    72.544371  \n",
       "Enterococcus faecalis        6550.505763  8745.103316  \n",
       "Leuconostoc citreum             3.259988    19.275630  \n",
       "Cutibacterium avidum          457.543378  4250.653466  \n",
       "Staphylococcus epidermidis   1187.660153   936.681889  \n",
       "Staphylococcus aureus        1124.427189   590.194971  \n",
       "Anaerococcus prevotii          73.851083  3366.910935  \n",
       "Staphylococcus lugdunensis      6.639770    10.595311  "
      ]
     },
     "execution_count": 101,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bins_df = df[['SRR492183','SRR492186','SRR492188','SRR492189', \n",
    "                        'SRR492190','SRR492193','SRR492194','SRR492197']]\n",
    "\n",
    "bins_df.rename(columns = {'SRR492183':'day1', 'SRR492186':'day2', 'SRR492188':'day3','SRR492189':'day4','SRR492190':'day5','SRR492193':'day6','SRR492194':'day7','SRR492197':'day8'}, inplace=True)\n",
    "bins_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAf8AAAEICAYAAABRZv0fAAAAOXRFWHRTb2Z0d2FyZQBNYXRwbG90bGliIHZlcnNpb24zLjMuMSwgaHR0cHM6Ly9tYXRwbG90bGliLm9yZy/d3fzzAAAACXBIWXMAAAsTAAALEwEAmpwYAABDhklEQVR4nO3deZwdVZ3+8c+ThC2EBAKyhSUICLIZIYAgQlBUEIZFQECUVSIKsswwCMNPBJUZEEZHQNDASNh3iIgsEUiCsiRsWVmEgSBLBMK+aCDdz++Pcy5Umnu7q5Puvrf7ft+86nXrnjp16lvVl5w6p05VyTYhhBBCaB796h1ACCGEEHpWVP4hhBBCk4nKP4QQQmgyUfmHEEIITSYq/xBCCKHJROUfQgghNJmo/ENoApJmSRrVg9s7SNJfemhboyQ93xPb6m0kDZdkSQPqHUtoLFH5h9DDJG0j6V5Jb0p6TdI9kjbvzm3a3tD2xO7cRgih94izwRB6kKTBwM3A94BrgMWBLwDz6hlX6H0kDbA9v95xhN4pWv4h9KxPAdi+0naL7X/YHm97OnzYXX6PpHNyz8Djkr5UWVnSEEn/K2mOpBck/UxS/8LywyQ9JultSY9K2jSnz5a0Q57vJ+kESf8n6VVJ10gampctKemynP6GpAckrVRtRwplVLa1x8ez1NyPD+PJ30+RdFmer3RVHyjpb5LmSjqpkHcpSWMlvS7pUWDzNhutGVflcoSks/L6z0jaqbB8qKSLJL2Yl48rLNtF0tR8XO6VtEmtP7KkrfOxezN/bp3T95X0YJu8x0q6Kc8vkWP7m6SXJP1G0lJ52ShJz0v6oaS/AxdV2W7/vP5cSU8DO7dZfnDh9/G0pO8Wls2U9C+F74vlckZ05ncReoeo/EPoWX8FWiRdLGknSctVybMl8DSwAvBj4IZK5QxcDMwH1gE+C3wF+A6ApL2BU4ADgMHArsCrVco/Ctgd2A5YFXgd+HVediAwBFgdWB44HPhHjX35P1KvxRDgVOAySauU3I8ytgHWA74EnCzp0zn9x8DaefpqjrmzcT2R4/o58L+SlJddCgwENgRWBH4JkE+ifgd8l3RcfgvcJGmJtkHnffwjcHbO+wvgj5KWB24C1pO0bmGVbwJX5PkzSCeII0h/42HAyYW8KwNDgTWB0R8/ZBwG7EL6bYwE9mqz/OW8fDBwMPDLygkicAnwrULerwFzbE+lc7+L0BvYjimmmHpwAj4NjAWeJ1XkNwEr5WUHAS8CKuSfAnwbWIl0eWCpwrL9gAl5/nbg6BrbnA3skOcfA75UWLYK8AHpMuAhwL3AJguxX1OB3Traj7bx5O+nAJfl+eGAgdXarLtvnn8a2LGwbDTwfCfieqqwbGDe1sr5OLQCy1Up43zgp23SngC2q5L328CUNmn3AQfl+cuAk/P8usDbOQ4B7wJrF9bbCngmz48C3geWbGdf7wIOL3z/St6/ATXyj6v8Zkgngm8Dg/P364Dj8/xC/y5iaswpWv4h9DDbj9k+yPZqwEakf3T/p5DlBed/cbNnc541gcWAObnr9Q1SC3TFnG91Uqu3I2sCNxbKeAxoIZ1cXEo6ibgqd33/XNJi1QqRdEChG/yNvC8rlNiPsv5emH8PGJTnVwWea1NuZ+L6sFzb7+XZQaTj95rt16vEsibwb5Uyc7mr19ifVdvGlL8Py/NXkE7aILX6x+U4PkE6CXiosI3bcnrFK7b/WWWbxW23d2x2knS/0kDTN0it+xUAbL8I3APsKWlZYCfg8rxq6d9F6B2i8g+hjmw/TuoF2KiQPKzQDQ2wBqkV/Ryp5b+C7WXzNNj2hjnfc6Su8I48B+xUKGNZ20vafsH2B7ZPtb0BsDWpi/iAtgVIWhO4ADgSWN72ssBMUuu1o/2A1MIdWFi2com4K+aQKt5iuZ2Jq5bngKG54qu27LQ2x2yg7Sur5H2RdLJQtAbwQp4fD6wgaQTpJKDS5T+X1JW+YWEbQ2wPKpTT0WtY2zs2SwDXA2eRepqWBW5hwWNzManrf2/gPtsvAJT9XYTeIyr/EHqQpPUl/Zuk1fL31UkVwP2FbCsCR+UBV3uTLhPcYnsOqeL4b0mDlQburS1pu7zehcBxkjZTsk6uDNv6DXBaZZmkT0jaLc9vL2ljpUGEb5EuB7RUKWNpUkX0Sl7vYBY8gam5H3nZVGDfvKzaten2XAOcKGm5fBx/0Mm4qsrH91bgvFz2YpK2zYsvAA6XtGU+tktL2lnSMlWKugX4lKRvShogaR9gA9JdHjiN0L8OOJN0/f5POb01b+eXklbM8Q+T9NWyB4Z0bI6StFoeT3JCYdniwBKkYzNfaaDjV9qsPw7YFDiaNAaAHEfZ30XoJaLyD6FnvU0acDZZ0rukSn8m8G+FPJNJ14LnAqcBe9muDNw7gPSP+KOkgXrXka5VY/vanP+KvJ1xpMqlrV+RxhmMl/R2jmHLvGzlXOZbpMsBk0jXqBdg+1Hgv0nXsl8CNiZ1GRe1tx8/IvVSvE4alHcF5Z1K6s5+hnQydGkn42rPt0kV2+OkwXHH5HIfJA2mOzfH/BRp/MDH5H3chfQ3fRU4HtjF9txCtiuAHYBrveDtej/MZd8v6S3gDtKgx7IuIHXPTwMeBm4oxPU2abDnNXkfvkn6HRRj/wepd2Ct4rqU/F2E3kMLXpILIdSTpIOA79jept6xhOYk6WTgU7a/1WHm0GvFQ35CCCEAH96meCipByT0YdHtH0IIAUmHkQY23mr77nrHE7pXdPuHEEIITSZa/iGEEEKTiWv+oVdbbPFhde+66t+vf8eZulmrW+sdQsNobZDezAEN8LtoaY278Yo+eP+FMs97aL+MuU+X+oEttsInF3lb3Sla/iGEEEKTiZZ/CCGEUFYf6U2Jyj+EEEIoq2V+x3l6gaj8QwghhJLcR8bXROUfQgghlNUalX8IIYTQXKLlH0IIITSZGPAXQgghNJlo+YcQQgjNxTHaP4QQQmgyfWTAX7c94U/SSZJmSZouaaqkLXP6MZIGLkK5YyXt1Yn8oyTdvLDbq5dGilvSbEkrtLN8uKRvFr6PlHR2z0QXQgg9yK3lpgbXLS1/SVsBuwCb2p6XK47F8+JjgMuA97pj26EuhgPfBK4AsP0g8GA9AwohhG7RRwb8dVfLfxVgru15ALbn2n5R0lHAqsAESRMAJJ0v6cHcS3BqpYDc2jxD0pQ8rVMof1tJ90p6utILIOlSSbsV1r9c0q7FoCQNlTQu90bcL2mTnD5I0kWSZuRle+b0/XLaTElnFMrZUdLDkqZJurODMt4prLeXpLF5fu9c7jRJtd6dPUjSdZIez/ujvO7Jkh7I648ppK8t6TZJD0n6s6T1c/rYfJwn5GO2naTfSXqsEk97+1tY/lNJRxe+n5b/pqcDX8g9PMcWey266biEEEJ99JGWf3dV/uOB1SX9VdJ5krYDsH028CKwve3tc96TbI8ENgG2q1TI2Vu2twDOBf6nkL4KsA2pd+H0nHYhcDCApCHA1sAtbeI6FXjE9ibAfwCX5PQfAW/a3jgvu0vSqsAZwBeBEcDmknaX9AngAmBP258B9q5VRgfH6GTgq7mMXWvk+Sypp2QD4JPA53P6ubY3t70RsFQ+DgBjgB/Y3gw4DjivUNZyeV+OBf4A/BLYENhY0oha+9smnv8FDgSQ1A/YF7gcOAH4s+0Rtn/ZZp0uPy6SRucTxgdbW9/toLgQQuhCLfPLTQ2uWyp/2+8AmwGjgVeAqyUdVCP7NyQ9DDxCqow2KCy7svC5VSF9nO1W248CK+VtTgLWkbQisB9wve22f4FtgEtz/ruA5fOJwg7Arwvxvw5sDky0/Uou53JgW+BzwN22n8l5X8urVSujPfcAYyUdBtR69+cU2887PU9yKql7HWB7SZMlzSBV1htKGkQ64blW0lTgt6STpIo/2DYwA3jJ9oxc7qxcbq39/ZDt2cCrkj4LfIV0IvVqB/vZ5cfF9hjbI22P7Ndv6Q6KCyGELtTaWm5qcN022t92CzARmJgrqQOBscU8ktYitVA3t/167vpdslhMjfl5xWIK85cC+5NapIdUCava+5Wd09u+o7nWu5ir5W0vvZj24b7ZPlxpEOTOwFRJI6pUpMX9bAEGSFqS1KIfafs5SafkcvsBb9geUSPuSlmtbcptJf0Oyp6qXggcBKwM/K5E/u44LiGEUBepauv9uqXlL2k9SesWkkYAz+b5t4Fl8vxg4F3gTUkrATu1KWqfwud9JTY9ltRNju1ZVZbfTTo5QNIo0riEt0iXKY4sxL8cMJl0GWIFSf1JvQmTchzb5RMXJA3Nq1UrA+AlSZ/O3eR7FJavbXuy7ZOBucDqJfYPPqoo5+bW/l55f98CnpG0dy5fkj5Tskza2d+2bgR2JPUU3J7Tin/TtnrquIQQQvfrI9f8u6vlPwg4R9KypBblU6RLAJCuS98qaY7t7SU9Qup6fprU5Vu0hKTJpJOU/TraqO2XJD0GjKuR5RTgIknTSXcbHJjTfwb8WtJMUgv7VNs3SDoRmEBqvd5i+/eQrjkDN+SK62Xgy9XKAG4gXQ+/GXgOmJmPDcCZ+QRJwJ3AtI72L+/jG5IuIHXfzwYeKCzeHzhf0v8DFgOu6kS5c2rtb5t87ysN1nzDH50CTwfmS5pGOgF7pLBKjxyXEELoEb2gS78MpcvAjUfSbFLX9txOrDOQVCluavvN7oqtmeUTnoeBvW0/We94Flt8WN1/wP371Rqy0XNae0FLo6e0Nsi/aQMa4HfR0kduS+sqH7z/Qq3LuaX986FxpX5gS262+yJvqzt120N+epqkHYDHgXOi4u8ekjYg9eLc2QgVfwgh9LiWD8pNDa5hH+9re3gn898BrNE90QSAfHfFJ+sdRwgh1E0f6fZv2Mo/hBBCaDh95BJbVP4hhBBCWdHyDyGEEJpMVP4hhBBCc3EvGMxXRlT+IYQQQllxzT+E+rth6LYdZ+pmO848rd4hMP/h2+odQrLkwHpHwEtHXV7vEAAYulm9I4B+Q5aqdwgALHbID+odQteJbv8QQgihyfSRln+fechPCCGE0O268K1+ko6VNEvSTElXSlpS0lBJf5L0ZP5crpD/RElPSXpC0lcL6ZtJmpGXnS2pw6cLRuUfQgghlNVFL/aRNAw4ivQY+41IrzDfl/Tekzttr0t6v8kJOf8GefmGpJernZdfwgZwPun9OevmaceOth+VfwghhFDW/PnlpnIGAEtJGgAMBF4EdgMuzssvBnbP87sBV9meZ/sZ0qPWt5C0CjDY9n1OL+u5pLBOTVH5hxBCCGWVbPlLGi3pwcI0eoFi7BeAs4C/AXOAN22PB1ayPSfnmQOsmFcZRnoLasXzOW1Ynm+b3q4Y8BdCCCGUVfJ6vu0xpFfYV5Wv5e8GrAW8AVwr6VvtFFntOr7bSW9XVP4hhBBCWV032n8H4BnbrwBIugHYGnhJ0iq25+Qu/Zdz/ueB1Qvrr0a6TPB8nm+b3q7o9g8hhBDK6rrR/n8DPidpYB6d/yXgMeAm4MCc50Dg93n+JmBfSUtIWos0sG9KvjTwtqTP5XIOKKxTU7T8QwghhLK6qOVve7Kk64CHgfnAI6TLBIOAayQdSjpB2DvnnyXpGuDRnP8I2y25uO8BY4GlgFvz1K6o/EMIIYSyyo/k75DtHwM/bpM8j9QLUC3/acDHHilq+0Fgo85su2m7/SW1SJpamE7oIP8oSVv3VHz1kh80MV3SsT2wrYmSRub5WyQt293bDCGERWKXmxpcM7f8/2F7RCfyjwLeAe4tu4KkAba77jSxm0laGdja9po9vW3bX+vpbYYQQqf1kWf7N23LvxZJsyWdKunh/LjE9SUNBw4Hjs29BF+Q9AlJ10t6IE+fz+ufImmMpPHAJZLWlHRnbk3fKWmNnG8lSTdKmpanrXP6v+ZHPc6UdEwhrgNyGdMkXVqrDEnDJc0srHecpFPy/FGSHs3lXFVl98cDKxb28bC8b9Pyvg7M5dTa90GSLsrHbbqkPXP6+fk+11mSTm3nuK8gaWlJf8zbnClpn0X5e4YQQpfqwsf71lMzt/yXkjS18P2/bF+d5+fa3lTS94HjbH9H0m+Ad2yfBSDpCuCXtv+SK/TbgU/n9TcDtrH9D0l/AC6xfbGkQ4CzSU9fOhuYZHsPpUc0DpK0GXAwsCXp3s3JkiYB7wMnAZ+3PVfS0Lydj5UBfPgc6CpOANayPa9GF/uuwM2VHhFJj9q+IM//DDgUOAf4VY19/xHpQRUb53UqsZxk+7Uc452SNrE9vUaMOwIv2t45lzGkbQalh2WMBvj+MiPZceA67exyCCF0oT7yYp9mrvzb6/a/IX8+BHy9Rp4dgA300fsTBktaJs/fZPsfeX6rQhmXAj/P818k3ZJBHrH5pqRtgBttvwsf3vf5BdIDG66zPTfnf62dMtqr/KcDl0saB4xrJ1/FRrnSX5Z0YnF7B/u+A+nZ0+SYXs+z38gV9gBgFWCDHEs1M4CzJJ1BOhH5c9sMxYdn/GHl/Rr/4loIoe9oaek4Ty/QzJV/e+blzxZqH6N+wFaFSh6AXCG+207Z7VVWtd7EpA7WK5rPgpdzlizM7wxsS2rh/0jShh2MSRgL7G57mqSDSOMeoPa+fyxOpftRjwM2t/26pLFtYlqA7b/mHpCvAf8labztn7QTYwgh9Jxe0KVfRlzzL+9tYJnC9/HAkZUvkkbUWO9ePmoN7w/8Jc/fSbo3E0n9JQ0G7gZ2V3row9LAHsCfc95vSFo+5x/aThkvka7bLy9pCWCXvLwfsLrtCcDxfNSab88ywBxJi+XYO9r3tunLAYNJJ0NvSloJ2Km9DUpaFXjP9mWk515v2kGMIYTQc/rINf9mrvyX0oK3+p3eQf4/AHtUBsORX8WYB7Y9ShoQWM1RwMGSpgPfBo7O6UcD20uaQbq8sKHth0mt7SnAZOBC24/YnkW6t3OSpGnAL9op4wPgJ3n9m4HHc97+wGU57yOka/ZvdLDPP8rl/KlQTmWfqu37z4Dl8kC9acD2tqfl7c0Cfgfc08E2Nwam5PEYJ+UyQwihMXTRK33rTe4F9yOGUEsjXPPfcebHnrnR4+Y/fFu9Q0iWHFjvCHjpqMvrHQIAQzerdwTQb8hS9Q4BgMUO+UG9QwBgiY2+XOvSamnvjTm21L85A0f/cpG31Z3imn8IIYRQVi/o0i8jKv8QQgihrBjtH0IIITSZaPmHEEIITSYq/xBCCKHJ9JFB8lH5hxBCCGVFyz+EEEJoMq19o+Uf9/mHXm3A4sPiBxxCKGX++y8s+n3+Zxxc7j7/H14U9/mHEEIIfYGj2z+EEEJoMn2k2z8q/xBCCKGsXvDc/jKi8g8hhBDKipZ/CCGE0GTmx+N9QwghhOYS3f4hhBBCk4lu/xBCCKG5xK1+IYQQQrOJln8IIYTQZPpI5d+v3gE0Cknv1DuGhSFpWUnf7+Zt3Js/h0v6ZnduK4QQGlpLS7mpwUXl3/stC3Rr5W976zw7HKha+UuKXqQQQp/nVpeaGl1U/u2QtLak2yQ9JOnPktbP6WMl7VXI905h/nhJMyRNk3R6Thsh6X5J0yXdKGm5nD5R0hmSpkj6q6Qv5PQlJV2Uy3lE0vY5fcOcd2oua13gdGDtnHamkjMlzczr79NebG32d6Uc37Q8bd1m/04HvpC3daykgyRdK+kPwHhJS0v6naQHcty75fUPknRuYTs3SxpVKTsfg4ck3SFpi3xcnpa06yL/EUMIoSu1utzU4KK11r4xwOG2n5S0JXAe8MVamSXtBOwObGn7PUlD86JLgB/YniTpJ8CPgWPysgG2t5D0tZy+A3AEgO2N8wnHeEmfAg4HfmX7ckmLA/2BE4CNbI/IMewJjAA+A6wAPCDp7pxWLbais4FJtveQ1B8Y1Gb5CcBxtnfJ2zoI2ArYxPZrkv4TuMv2IZKWBaZIuqPW8cqWBiba/qGkG4GfAV8GNgAuBm5qu4Kk0cBoAPUfQr9+S3ewiRBC6CIx2r9vkzQI2Bq4VvrwzYxLdLDaDsBFtt8DyBXiEGBZ25NynouBawvr3JA/HyJ1qwNsA5yTy3hc0rPAp4D7gJMkrQbckE9K2sawDXCl7RbgJUmTgM2B7drGViX+LwIH5OUtwJsd7C/AnwplfQXYVdJx+fuSwBodrP8+cFuenwHMs/2BpBl8dDwWYHsM6cQsXukbQuhZvaBVX0ZU/rX1A96otKjbmJ+Xo1T7Lp7TBXT2lzEvf7bw0d+j6nugbV8haTKwM3C7pO8AT7fJVusd0gsTWxnvttnGnrafWGDD0mYseIlpycL8B7YrcbWSj4ft1hhHEEJoOH2k8o9r/jXYfgt4RtLekCp5SZ/Ji2cDm+X53YDF8vx44BBJA/M6Q22/CbxeuZ4PfBuo9ALUcjewfy7jU6TW8xOSPgk8bftsUnf4JsDbwDJt1t1HUn9JnwC2BaZUi63Kdu8EvpeX95c0uM3ytttq63bgB/mECEmfzemzgRGS+klaHdiig/0PIYSG5JbWUlOji8r/IwMlPV+Y/pVUAR8qaRowi1TRA1wAbCdpCrAlufVr+zZSpfygpKlApfv7QOBMSdNJ195/0kEs5wH9c9f31cBBtucB+wAzc9nrA5fYfhW4Jw/wOxO4EZgOTAPuAo63/fd2Yis6Gtg+b/chYMM2y6cD8/NgwGOrrP9T0onQdEkz83eAe4BnSN36ZwEPd7D/IYTQmPrIgD991OMaQu8T1/xDCGXNf/+FWpdFS3vz4B1K/Zsz5KI7Fnlb3SmuqYYQQghl9YJWfRnR7R9CCCGU1VpyKkHpCa3XSXpc0mOStpI0VNKfJD2ZP5cr5D9R0lOSnpD01UL6ZvkZLk9JOrsy7qo9UfmHEEIIJXl+a6mppF8Bt9len/RslsdIz1O50/a6pEHYJwBI2gDYlzQWa0fgvPw8FoDzSc8+WTdPO3a04Q4rf0mfl7R0nv+WpF9IWrPsnoUQQgh9Rhe1/PPdVNsC/wtg+33bb5AGll+cs11MejgbOf0q2/NsPwM8BWwhaRVgsO378m3TlxTWqalMy/984L18m9vxwLO58BBCCKGplH22v6TRkh4sTKPbFPVJ4BXgovw49AtzQ3sl23MA8ueKOf8w4LnC+s/ntGF5vm16u8pU/vPz2cRupEfL/or27/UOIYQQ+qaSLX/bY2yPLExj2pQ0ANgUON/2Z0m3jJ/QzparXcd3O+ntKlP5vy3pROBbwB/zNYbFOlgnhBBC6HO68K1+zwPP256cv19HOhl4KXflkz9fLuRfvbD+asCLOX21KuntKnOr3z6k17geavvvktYAziyxXgjdbpsVP13vEPjLy4/VO4TQgAYu1tGrQLrf4v0b427uK5batN4hdJ0uenhfrk+fk7RefiT6l4BH83Qg6S2qBwK/z6vcBFwh6RfAqqSBfVNst0h6W9LngMmk97Oc09H2O/xl2P478IvC978R1/xDCCE0Ic/v0uJ+AFTe0vo0cDCpR/4aSYcCfwP2BrA9S9I1pJOD+cAR+QVskB7LPhZYCrg1T+3qsPKX9HXgDNKgA+XJtts+9z2EEELo09yFj+23PRUYWWXRl2rkPw04rUr6g8BGndl2mT6hnwP/Yjv6NkMIITS3xn9nTyllKv+XouIPIYQQurblX09lKv8HJV0NjOOjd89j+4buCiqEEEJoRM1U+Q8G3gO+UkgzEJV/CCGEpuKWhn5ZX2llRvsf3BOBhBBCCI2uz7f8JR1v++eSzqHK04JsH9WtkYUQQggNxq19v+VfGeT3YE8EEkIIITS6Pt/yt/2H/HkxfPgGItt+u4diCyGEEBqK3Tda/mVe6TtS0gxgOjBT0jRJm3V/aF1P0sqSrpL0f5IelXSLpE91sM4xkgYWvt8iaVlJwyXN7IKYhkv65kKue++ibr8r1IpD0lhJe/V0PCGE0F3cWm5qdGVe7PM74Pu2h9teEzgCuKh7w+p6kgTcCEy0vbbtDYD/AFbqYNVjgA8rf9tfy+9c7irDSe9OKC2/XAnbW3dhHAutUeIIIYTu1tqiUlOjK/VWP9t/rnyx/RegN3b9bw98YPs3lQTbU23/WdIoSTdX0iWdK+kgSUeRXqAwQdKEvGy2pBVy1gGSLpY0XdJ1lR4CSSdLekDSTElj8okHktaRdEfuPXlY0tqklzd8QdJUScdK6i/pzLz+dEnfzeuOkjRB0hXAjJz2TmHZx+IvxPufku7L75TeVNLtuffj8GoHStI4SQ9JmlV5B7Wk70n6eSHPQXkwaDEO5W0/KumPfPQe6gWOW+5NmpjnT8nHcHzO83VJP5c0Q9JtkuINkiGEhuFWlZoaXc3KP1cSmwJTJP02VzDbSToPmNhjEXadjYCHOrOC7bNJr0bc3vb2VbKsB4yxvQnwFvD9nH6u7c1tb0R60cIuOf1y4Ne2PwNsDcwhvb/5z7ZH2P4lcCjwpu3Ngc2BwyStldffAjgp91p0xnO2twL+THr5w17A54Cf1Mh/iO3NSM+cPkrS8qTXTX69kGcf4Oo26+1BOiYbA4flfSxjbWBnYDfgMmCC7Y2Bf+T0BUganU9kHnzx3RdKbiKEEBZdX6n82xvt/99tvv+4MF/qZcVN4Dnb9+T5y4CjgLOA7SUdT7pcMBSYlVu6w2zfCGD7nwC5U6DoK8AmhWvlQ0ivbnyf9PrGZxYizpvy5wxgUB60+bakf0patspljKMk7ZHnVwfWtX2/pKeVXhv5JKmSv6fNetsCV+Y3Tb0o6a6S8d1q+4M8tqQ/cFsh3uFtM9seA4wBGLXaDvFbDCH0GPeRf3HaG+1fraXbm80itXirmc+CvSBLliyz7c/AkpYEzgNG2n5O0im5vLKnggJ+YPv2BRKlUcC7NdbpKP7KY5lbC/OV7wv8BvJ2dgC2sv1ePmmplHc18A3gceBGu+r/BrX+1yjGWDU+262SPiiU+7H4QgihnnpDq76MMtf8+4q7gCUkHVZJkLS5pO2AZ4ENJC0haQgLvk7xbWCZGmWuIWmrPL8f8Bc+qtjmShpEPuGw/RbwvKTd87aXyGME2pZ/O/C9yrVuSZ+StHQH+9Ze/J01BHg9V/zrky4PVNwA7E7a17Zd/gB3A/vmcQurkMZZVMwGKneJ7LkI8YUQQt3YKjU1uqap/HNrcg/gy3mw2yzgFOBF288B15BuZ7wceKSw6hjg1sqAvzYeAw6UNJ3UvX9+7kK/gNRlPQ54oJD/26Qu9enAvcDKeZvz8yDAY4ELgUeBh5VuJfwtHbR+O4i/s24jDWScDvwUuL+wnddzbGvanlJl3RtJlwRmAOcDkwrLTgV+JenPQMsixBdCCHXT0qJSU6NT9Z7bEHqHRrjm/5eX443X4eMGLrZEvUNg8f6NcdXsiqU2rXcIAHzlpasWuVZ+Yv2dSv2bs97jtzb0GUCZh/wcIWnZwvflJH2/nVVCCCGEPqmvjPYv0+1/WHE0eO76Pax29hBCCKFvsstNja5Mn1A/SaqMwFZ6utzi3RtWCCGE0Hh6Q6u+jDKV/+3ANZJ+Q7qN63A+ug87hBBCaBotrX1jnHyZyv+HwHeB75HuQR9PGpEeQgghNJXe0KVfRoeVv+1W0m1b53d/OCGEEELjau0F9/CXUbPyl3SN7W/kR65+7FwnP88+hBBCaBq94QE+ZbTX8j86f+7STp4QQgihafT5bn/bc/LnswCSBreXP4R6eK/1g3qHwBqDV+w4Uzdbqn9j3IDT2gD/MvZXYwzIWnnxIfUOgVc/eKfeIQBwZMtT9Q4BgL92QRl9vtu/Ir9P/iek16tW/s828MlujCuEEEJoOM002v84YEPbc7s7mBBCCKGR1b9vq2uUqfz/D3ivuwMJIYQQGl3TdPsDJwL3SppM4V3wto/qtqhCCCGEBtQMo/0rfgvcRXpNa2v3hhNCCCE0rr5SCZap/Ofb/tdujySEEEJocKZ5Wv4TJI0G/sCC3f6vdVtUIYQQQgOa30Td/t/MnycW0uJWvxBCCE2naVr+ttfqiUBCCCGERtc01/wlLUZ6o9+2OWki8Fvb9X+0WgghhNCD+krLv8yjis4HNgPOy9NmdMEb/iSdJGmWpOmSpkraMqcfI2ngIpQ7VtJencg/StLNC7u93kTShZI2qJJ+kKRz67VNSYdLOqA7th9CCF2pteTU6Mpc89/c9mcK3++SNG1RNippK9ILgza1PU/SCkDl4eTHAJcRDxbqcra/0xXlSBpge35XbdP2bxY9qhBC6H4tTdTyb5G0duWLpE8CLYu43VWAubbnAdiea/tFSUcBq5LuMJiQt3e+pAdzL8GphThmSzpD0pQ8rVMof1tJ90p6utILIOlSSbsV1r9c0q7FoCQNlTQu90bcL2mTnD5I0kWSZuRle+b0/XLaTElnFMrZUdLDkqZJurODMt4prLeXpLF5fu9c7jRJd1c7iJL+XdIDubxTc9pwSY9LujinX1fpSZE0UdLIPH+wpL9KmgR8vlDmJyRdn8t9QNLnc/opksZIGg9ckr9fLGl8/lt8XdLP8/7dli8Xld3mKZKOy/NHSXo0x35Vtf0OIYR6aVW5qSxJ/SU9UumBzvXQnyQ9mT+XK+Q9UdJTkp6Q9NVC+mb5396nJJ0tqcMIylT+/06qjCfmf7TvAv6t/K5VNR5YPVcE50naDsD22cCLwPa2t895T7I9EtgE2K5SIWdv2d4COBf4n0L6KsA2pN6F03PahcDBAJKGAFsDt7SJ61TgEdubAP8BXJLTfwS8aXvjvOwuSasCZwBfBEYAm0vaXdIngAuAPXOPyd61yujgGJ0MfDWXsWvbhZK+AqwLbJG3v5mkyriM9YAxeTtvAd9vs+4qeV8/D3wZKHbL/wr4pe3NgT3zcavYDNjNduUOkLWBnYHdSL01E2xvTHoJ1M6d2GbRCcBnc+yHV8sgaXQ+IXzw5ffm1CgmhBC6XisqNXXC0cBjhe8nAHfaXhe4M39H6fLpvsCGwI7AeZL653XOB0aT6oR18/J2dVj5274zF3ZUntazPaHcPtUs8x1SRTIaeAW4WtJBNbJ/Q9LDwCOknS5WGlcWPrcqpI+z3Wr7UWClvM1JwDqSVgT2A66v0nW9DXBpzn8XsHw+UdgB+HUh/teBzYGJtl/J5VxOGhT5OeBu28/kvJXnIVQroz33AGMlHQb0r7L8K3l6BHgYWJ/0dwJ4zvY9ef6yvF9FWxZifx+4urBsB+BcSVOBm4DBkpbJy26y/Y9C3lvzwM8ZOcbbcvoMYHgntlk0Hbhc0reAqpcWbI+xPdL2yBUHrlKjmBBC6HouOZUhaTVSQ6nYyNoNuDjPXwzsXki/yva8XL88BWyRG1aDbd9n26RG6+50oMw1f0gV9fCc/zOSsH1J+6u0z3YL6c6BiZJmAAcCY4t5JK1Feqvg5rZfz13iSxaLqTE/rzBfPAW7FNifdPZ0SJWwqp2uOae3/XvWOrWrlre99GLah/tm+3ClQZA7A1MljbD9apvy/sv2bxfYiDS8ynY62m5RP2CrNpU8uRfp3TZ5K5dtWiV9kH94kMa7VPttlfl/YmfSSdSuwI8kbVh2fEEIIXS3soP5lB6ON7qQNMb2mDbZ/gc4HlimkLaS7TkAtufkBivAMOD+Qr7nc9oHeb5ters6bPlLuhQ4i9R63DxPIztar4My15O0biFpBPBsnn+bjw7EYFKF86aklYCd2hS1T+HzvhKbHksaUIjtWVWW3006OUDSKNK4hLdIlymOLMS/HDCZdBlihdz1sh8wKcexXT5xQdLQvFq1MgBekvRpSf2APQrL17Y92fbJwFxg9Tax3g4cImlQzj+s8CNZQ2lQJTmuv7RZdzIwStLy+dr83oVlbeMcUeU4LYz2tlnZVj9g9dyzdDywLDCoi7YfQgiLrFUqNRV7KPO0QMUvaRfgZdsPldx0e43TauntKtPyHwlsUGjVdYVBwDmSliV17T7FR2dIY4BbJc2xvb2kR4BZwNOkrvCiJZTeNtiPVMm1y/ZLkh4DxtXIcgpwkaTppLsNDszpPwN+LWkmabDjqbZvkHQiMIF08G+x/Xv48IzvhlyZvUy6xv2xMoAbSNdzbgaeA2byUWV3Zj5BEum6zwJ3WNgeL+nTwH25Vf4O8K1c9mPAgZJ+CzxJm1sz89nkKaQTlTmkywaVSwtH5Tink34fd1Pj2ntndLDNiv7AZflSi0hjD95Y1G2HEEJXWdTR7gWfB3aV9DVSr+9gSZeRGoSr5H8zVyHVIZBa9MVG4GqkMXLP5/m26e1SR3W6pGuBoyrdEI1C0mxgpO25nVhnIOl69Ka23+yu2Oopd/vfbHujesfSE7ZYdbuuPCldKHPff6veIbBU/8U7ztQDWru0jbBw+qvMOObut/LiQ+odAq9+8E7HmXrAuy3zOs7UA/76yoOLfJ/elavuX+pHvt+Ll5feVu5pPs72LpLOBF61fbqkE4Chto+XtCFwBWmQ96qkRuG6tlskPQD8gNTDegtwju22A9oXUKblvwLwqKQpLPhin4+NQG9kknYAfgf8oq9W/CGEELpXJ0fyL4zTgWskHQr8jXyJ1PYsSdcAj5J6zI/IY+cgPYV3LLAUcGue2lWm8j+ls5H3BNvDO5n/DmCN7ommcdieDTRFqz+EEHpad/Rt2Z5IGgBPHtj9pRr5TgNOq5L+IJ38d7/Mi30mSVqT1L1wR+46r3brWQghhNCndeYBPo2szGj/w4DrgMotZcOoPWAuhBBC6LP6yrP9y4yMOYI0KvEtANtPAiu2u0YIIYTQB7Wo3NToylzzn2f7/cqjgiUNoHsue4QQQggNrTe06sso0/KfJOk/gKUkfRm4FvhD94YVQgghNJ6+0u1fpuV/AnAo6f7475LuIbyw3TVC6CEPz32q3iGEUNXjPFfvENhgaGPc4PT2B33nDe3uBV36ZZQZ7d9KekvdBd0fTgghhNC4ekOrvoya3f6SdpN0ROH7ZElP5+ljz2UPIYQQ+rqWklOja++a//GkV7pWLEF6qc8ouuBZ7yGEEEJv06pyU6Nrr9t/cdvFi1Z/yU8eelXS0t0cVwghhNBw+kq3f3uV/3LFL7aPLHz9RPeEE0IIITSuvlL5t9ftPzk/3W8Bkr4LTOm+kEIIIYTG5JJTo2uv5X8sME7SN0nvXgfYjHTtf/dujiuEEEJoOL3hen4ZNSt/2y8DW0v6IrBhTv6j7bt6JLIQQgihwfSGkfxllLnP/y4gKvwQQghNr7VXdOp3rMwT/kIIIYRAcwz46/UknSRplqTpkqZK2jKnHyNp4CKUO1bSXp3IP0rSzQu7vRBCCI2hGQb89WqStgJ2ATa1PU/SCsDiefExwGVA33ngdAOSNMD2/HrHEUIIXSVa/o1vFWCu7XkAtufaflHSUcCqwARJEwAknS/pwdxLcGqlAEmzJZ0haUqe1imUv62ke/PjjvfK+S+VtFth/csl7VoMStJQSeNyb8T9kjbJ6YMkXSRpRl62Z07fL6fNlHRGoZwdJT0saZqkOzso453CentJGpvn987lTpN0d9sDmMu7M29nRmXfJA2XNLOQ7zhJp+T5iZL+U9Ik4GhJm0maJOkhSbdLWqWQb2SeX0HS7Dy/YT7WU/M+rNvxnzqEEHrGfLnU1Oj6bMsfGA+cLOmvwB3A1bYn2T5b0r8C29uem/OeZPs1Sf2BOyVtYnt6XvaW7S0kHQD8D6k3AdLJxTbA+qTHIF9HetvhscDvJQ0BtgYOzPkqTgUesb17vpPiEmAE8CPgTdsbA0haTtKqwBmkWyxfB8ZL2h24h/SipW1tPyNpaC77Y2V0cIxOBr5q+wVJy1ZZ/k9gD9tv5Z6T+yXdVCVfW8va3k7SYsAkYDfbr0jaBzgNOKSddQ8HfmX7ckmLA/3bZpA0GhgNoP5D6NcvHjgZQugZjV+tl9NnW/623yFVmqOBV4CrJR1UI/s3JD0MPEK6rXGDwrIrC59bFdLH2W61/SiwUt7mJGAdSSsC+wHXV+n23ga4NOe/C1g+nyjsAPy6EP/rpHcpTLT9Si7ncmBb4HPA3bafyXlfy6tVK6M99wBj88OcPlbJAgL+U9J00gnUsMq+duDq/LkesBHwJ0lTgf8HrNbBuvcB/yHph8Catv/RNoPtMbZH2h4ZFX8IoSe1lpwaXV9u+WO7BZgITJQ0g9QKH1vMI2kt4Dhgc9uv5y7xJYvF1JifVyymMH8psD+wL9VbuNUeEeGc3vakstbjJKrlbS+9mPbhvtk+PA+C3BmYKmlEfn9Dxf6kRzlvZvuD3DW/JDCfBU8ci8cL4N1CPLNsb8XHFcsoxnSFpMk5ptslfSeeLRFCaBR95Va/Ptvyl7Rem+vFI4Bn8/zbwDJ5fjCpsnpT0krATm2K2qfweV+JTY8lDSjE9qwqy+8mVapIGkUal/AW6TLFh+9PyF32k4Ht8jXx/qTehEk5ju3yiQuFbv9qZQC8JOnTkvoBexSWr217su2TgbnA6m1iHQK8nCv+7YE1K+UBK0paXtISfHQppK0ngE8oDb5E0mKSKg+Mmk3qmQH48M4JSZ8EnrZ9NulyyiY1yg4hhB4Xo/0b3yDgnHwtez7wFPk6MTAGuFXSHNvbS3oEmAU8TeoKL1oit0T7kSrfdtl+SdJjwLgaWU4BLspd6e+ReiMAfgb8Og+kawFOtX2DpBOBCaRW9C22fw8fXve+IVfoLwNfrlYGcANwAnAz8BwwMx8bgDPzCZKAO4FpbWK9HPiDpAeBqcDjeR8/kPQT0snJM5X0KsfifaXBkGfnSxsDSOMmZgFnAddI+jYLPkRqH+Bbkj4A/g78pMZxDCGEHtcbuvTLkN0bzlHqI3dzjywMDCyzzkBgBukWwze7K7aQDFh8WPyAQ6hhg6Fr1DsEAF75Z2P8UzjnjUcX+cn8xw7ft9S/Ob+cfVVDvwWgz3b714OkHUit4HOi4g8hhL4nBvw1AdvDO5n/DqAxTrVDCCF0OfeKK/odi8o/hBBCKKk3tOrLiMo/hBBCKKmv3OoXlX8IIYRQUt+o+qPyDyGEEEqb30eq/6j8QwghhJJiwF8IIYSG9vjrz9U7BADOXGlUvUPoMjHgL4QQQmgy0fIPIYQQmky0/EMIIYQm09JHHokflX8IIYRQUl+5zz+e7R9CCCGU5JL/dUTS6pImSHpM0ixJR+f0oZL+JOnJ/LlcYZ0TJT0l6QlJXy2kbyZpRl52tqQOXyoUlX8IIYRQUhe+2Gc+8G+2Pw18DjhC0gakV7DfaXtd0qvWTwDIy/YFNgR2BM6T1D+XdT7plfXr5mnHjjYelX8IIYRQUisuNXXE9hzbD+f5t4HHgGHAbsDFOdvFwO55fjfgKtvzbD8DPAVsIWkVYLDt+2wbuKSwTk1R+YcQQgglle32lzRa0oOFaXStMiUNBz4LTAZWsj0H0gkCsGLONgwoPrjh+Zw2LM+3TW9XDPgLIYQQSio72t/2GGBMR/kkDQKuB46x/VY7l+urLXA76e2Kln8IIYRQUld1+wNIWoxU8V9u+4ac/FLuyid/vpzTnwdWL6y+GvBiTl+tSnq7mrLyl7SHJEtav96xNDtJIyR9rfB9V0mVAS6HSzqgftGFEMKCumrAXx6R/7/AY7Z/UVh0E3Bgnj8Q+H0hfV9JS0haizSwb0q+NPC2pM/lMg8orFNTs3b77wf8hTRy8pTu3JCkAbbnd+c2epqk/rZbuqi4EcBI4BYA2zeRfuTY/k0XbSOEELpEFz7e9/PAt4EZkqbmtP8ATgeukXQo8DdgbwDbsyRdAzxKulPgiMK/w98DxgJLAbfmqV1N1/LP11c+DxxKqvwr6aMkTZR0naTHJV1euVdS0smSHpA0U9KYQvrakm6T9JCkP1d6EiSNlfQLSROAM3Lr9n5J0yXdWLlvU9I6ku6QNE3Sw7k8STozb2uGpH0KMR6f06ZJOr2dMkZJurmw3rmSDsrzp0t6NMdyVpXjc4qkSyXdle8zPaxwfCZIuoL0Y+2f43wgl/XdnO/qNi35sZL2lLSkpIty/I9I2l7S4sBPgH0kTZW0j6SDJJ1biOW4Rf+rhxBC1+jC0f5/sS3bm9gekadbbL9q+0u2182frxXWOc322rbXs31rIf1B2xvlZUfmUf/tasaW/+7Abbb/Kuk1SZtWbrcgjbbckHS95B7SScJfgHNt/wRA0qXALsAfSIM5Drf9pKQtgfOAL+ayPgXsYLtF0nTgB7YnSfoJ8GPgGOBy4HTbN0paknQy9nVSa/gzwArAA5Luzmm7A1vafk/S0LydamUUrwt9KK+zB7C+bUtatsYx2oR03+nSwCOS/pjTtwA2sv2M0sjVN21vLmkJ4B5J44GrgH2AW3Ll/iXSWekRALY3zidJ4/MxOhkYafvIHONBNWIq7sdo0j2tqP8Q+vVbuqNVQgihS5SoV3uFpmv5k7r8r8rzV+XvFVNsP2+7FZgKDM/p20uaLGkGqXLfMPcgbA1cm7tsfgusUijr2lzxDwGWtT0pp18MbCtpGWCY7RsBbP/T9nvANsCVtltsvwRMAjYHdgAuynmw/Vo7ZdTyFvBP4EJJXwdq5f297X/YngtMIFX6lePzTJ7/CnBA3vfJwPKka1C3Al/MJwQ7AXfb/kfer0tznI8Dz5Iq/06zPcb2SNsjo+IPIfSkFlxqanRN1fKXtDyp8t5IkoH+gCUdn7PMK2RvAQbk1vR5pNbpc5JOASot7Ddsj6ixuXc7Cmch0tv+omrlnc+CJ3ZLAtieL2kLUmt8X+BIPuqpKGq7ncr34j6J1Jtx+8cClSYCXyX1AFzZQawhhNBrxLP9e6e9gEtsr2l7uO3VgWdIrdJalsyfc3Nrfy8A228Bz0jaG9LITUmfabuy7TeB1yV9ISd9G5iU139e0u55/SUkDQTuJl0D7y/pE8C2wBRSN/khOQ+ShrZTxrPABvn7EFJlXxnvMMT2LaTLDiNq7PNu+Rr98sAo4IEqeW4Hvqd0qwqSPiWp0gy/CjgY+ELOR96v/St5gTWAJ4C3gWVqxBFCCA3Fdqmp0TVb5b8fcGObtOuBb9ZawfYbwAXADGAcC1aE+wOHSpoGzCI9frGaA4Ez87X/EaRBbpBOBI7K6fcCK+f4pgPTgLuA423/3fZtpFHwD+au9uNqlWH7OeCaXM7lwCM57zLAzTnvJODYGvFOAf4I3A/81Ha1e0YvJI06fVjSTNJlj0pP0njSScsdtt/PaecB/fOlk6uBg2zPI11W2KAy4K9GPCGE0BC68j7/elJvOEMJPSdf1njH9sfuBGhEAxYfFj/gEGro1/HL3XrEmSuNqncIABz9t8sW+YCMWm2HUv/mTHz+jsY4+DU01TX/EEIIYVGUfbxvo4vKPyzA9in1jiGEEBpVb+jSLyMq/xBCCKGkqPxDCCGEJtNXxslF5R9CCCGUFC3/EEIIocl04Yt96ioq/xBCCKGkFpd5YW/ji8o/hBC6QSPcY9/aINen/+3vE+odAgBHd0EZcc0/hBBCaDJxzT+EEEJoMnHNP4QQQmgyjXIpZVFF5R9CCCGUFC3/EEIIocnEaP8QQgihyUS3fwghhNBkots/hBBCaDLR8g8hhBCaTLT8QwghhCbT4pZ6h9Al+vX0BiWdJGmWpOmSpkraMqcfI2ngIpQ7VtJencg/StLNC7u9epE0XNLMLizvna4qaxFi+ImkHeodRwghdMR2qanR9WjLX9JWwC7AprbnSVoBWDwvPga4DHivJ2MK9Wf75HrHEEIIZfSVx/v2dMt/FWCu7XkAtufaflHSUcCqwARJEwAknS/pwdxLcGqlAEmzJZ0haUqe1imUv62keyU9XekFkHSppN0K618uaddiUJKGShqXeyPul7RJTh8k6SJJM/KyPXP6fjltpqQzCuXsKOlhSdMk3dlBGe8U1ttL0tg8v3cud5qku9s7mJIOknRu4fvNkkbl+UMl/VXSREkXVPJJWkvSfZIekPTTwroL9IRIOlfSQYVjfmretxmS1s/pS0v6XS7rkcpxznHdIOk2SU9K+nlO7597aGbmco7N6R/22kg6XdKj+Vid1d7+hxBCT4uW/8IZD5ws6a/AHcDVtifZPlvSvwLb256b855k+zVJ/YE7JW1ie3pe9pbtLSQdAPwPqTcB0snFNsD6wE3AdcCFwLHA7yUNAbYGDsz5Kk4FHrG9u6QvApcAI4AfAW/a3hhA0nKSVgXOADYDXgfGS9oduAe4ANjW9jOShuayP1ZGB8foZOCrtl+QtGyHR7SKHOOPgE2Bt4G7gGl58a+A821fIumIThQ71/amkr4PHAd8BzgJuMv2ITnWKZLuyPlHAJ8F5gFPSDoHWBEYZnujHOcC+5eP2R7A+rZda/8ljQZGA6j/EPr1W7oTuxFCCAuvr4z279GWv+13SJXmaOAV4OpK67KKb0h6GHgE2BDYoLDsysLnVoX0cbZbbT8KrJS3OQlYR9KKwH7A9bbnt9nWNsClOf9dwPL5RGEH4NeF+F8HNgcm2n4ll3M5sC3wOeBu28/kvK/l1aqV0Z57gLGSDgP6d5C3li2ASbZfs/0BcG1h2ef56Phd2okyb8ifDwHD8/xXgBMkTQUmAksCa+Rld9p+0/Y/gUeBNYGngU9KOkfSjsBbbbbxFvBP4EJJX6fGJSDbY2yPtD0yKv4QQk9yyf8aXY8P+LPdYnui7R8DRwJ7ts0jaS1S6/JLtjcB/kiqWD4spsb8vGIxhflLgf2Bg4GLqoRV7cXbzult/4q1XtJdLW976cW0D/fN9uHA/wNWB6ZKWr7G9gDms+DfsFJORy8SrxZPrbIqKse2hY96jATsaXtEntaw/Vib/B+uk098PkM6UTiC1CvzUVDpZGoL4Hpgd+C2DvYjhBB6VItbS02Nrkcrf0nrSVq3kDQCeDbPvw0sk+cHA+8Cb0paCdipTVH7FD7vK7HpsaQBhdieVWX53aSTA/I187m23yJdpjiyEP9ywGRgO0kr5EsS+wGTchzb5ROXShc2NcoAeEnSpyX1I3V1V5avbXtyHgQ3l3QSUMtsYISkfpJWJ1WcAFNyLMtJGsCCJ1j3APvm+f0L6c8CG0haIvd6fKmd7VbcDvxAknLsn20vs9IAz362r+ejyxLF5YOAIbZvIf29RpSIIYQQekxc8184g4Bz8rXc+cBT5Gu3wBjgVklzbG8v6RFgFqmr+J425SwhaTLp5GW/jjZq+yVJjwHjamQ5BbhI0nRSV/OBOf1nwK+Vbq1rAU61fYOkE4EJpJbvLbZ/Dx9ei74hV+gvA1+uVgapC/0E4GbgOWBmPjYAZ+YTJAF38tG1+mruAZ4BZuQyHs77+4Kk/ySdqLxI6nZ/M69zNHCFpKNJLezKMXpO0jXAdOBJ0uWWjvyUNOZiej4BmM1H4y+qGUY6zpWTzhPbLF+GNDZjSdL+H1sihhBC6DF95Zq/esMZSpGk2cDIwsDAMusMJFWQm9p+s6P8fYGkQbbfyS3/G4Hf2b6x3nF1tQGLD+tdP+DQNPqpo6tv3a+vVFRdZf77LyzyH2W5QeuUOqivv/NU/X8A7ejxa/49TenhMY8D5zRLxZ+dkgfizST1DoyrazQhhNAHtOJSU6PrdS3/EIqi5R8aVbT8G09XtPwHL/3JUgf1rXefrv8PoB3xbP8QQgihpN4wkr+MqPxDCCGEkvpKb0pU/iGEEEJJfeVSeVT+IYQQQkm94el9ZUTlH0IIIZQULf8QQgihyfSVa/5xq19oepJG2x7T7DE0ShyNEEOjxNEIMTRKHI0QQ1/S5x/yE0IJozvO0u0aIQZojDgaIQZojDgaIQZojDgaIYY+Iyr/EEIIoclE5R9CCCE0maj8Q0hvlKy3RogBGiOORogBGiOORogBGiOORoihz4gBfyGEEEKTiZZ/CCGE0GSi8g8hhBCaTFT+oc+TdIqk4xZivb0lzZLUKmlknWI4U9LjkqZLulHSsnWK46c5hqmSxktatadjKKx/nCRLWmFhy1iUOPJ6L+RjMVXS13o6hrzuDyQ9kX+jP1/YGBYlDklXF47DbElT6xDDCEn35xgelLTFwsbQTKLyD6G2mcDXgbvrGMOfgI1sbwL8FTixTnGcaXsT2yOAm4GT6xGEpNWBLwN/q8f2C35pe0SebunpjUvaHtgN2MT2hsBZPR0DgO19KscBuB64oQ5h/Bw4Ncdwcv4eOhCVf+iTJJ2UW0V3AOvltMMkPSBpmqTrJQ2UtIykZyQtlvMMzi2YxWw/ZvuJOscw3vb8XOT9wGp1iuOtQpFLQ+febtIVMeSifgkc39ntd0McC62LYvgecLrteQC2X67nsZAk4BvAlXWIwcDgXOQQ4MXOHotmFJV/6HMkbQbsC3yW1HLfPC+6wfbmtj8DPAYcavttYCKwc86zL3C97Q8aMIZDgFvrFYek0yQ9B+xPJ1r+XRWDpF2BF2xP68wx6Oo48vcjlS6D/E7ScnWI4VPAFyRNljRJ0uZ0Qjf8Pr8AvGT7yTrEcAxwZv5tnkX9esd6laj8Q1/0BeBG2+/lFutNOX0jSX+WNINUgW2Y0y8EDs7zBwMXNVoMkk4C5gOX1ysO2yfZXj3HcGRPxiBpIHASi3a5oauOxfnA2sAIYA7w33WIYQCwHPA54N+Ba3Lru6fjqNiPTrb6uzCG7wHH5t/mscD/djKOphSVf+irqnULjwWOtL0xcCqwJIDte4DhkrYD+tue2UgxSDoQ2AXY3wv3YI6uPhZXAHv2cAxrA2sB0yTNJl3+eFjSyj0cB7Zfst1iuxW4AOjsALOu+Hs8T2oh2/YUoBXo7ADIrvp9DiC13K/u5Pa7KoYD+WiswbV0/u/RlKLyD33R3cAekpaStAzwLzl9GWBOvk64f5t1LiG1XLqi1d9lMUjaEfghsKvt9+oYx7qF5bsCj/dkDLZn2F7R9nDbw0mV36a2/96TcQBIWqWwfA/SwNAejQEYB3wxx/MpYHFgbh3iANgBeNz2853YflfG8CKwXZ7/IlD60kMziyf8hT4pd5MfADxLqigeBd4lDRZ7FpgBLGP7oJx/ZeAZYBXbb+S0PYBzgE8AbwBTbX+1h2N4ClgCeDUXe7/tw+twLK4nDchqzescbvuFnoyhTXmzgZG2O1PhddWxuJTU5W9gNvBd23N6OIbFgd/lON4HjrN9V08fi5w+lvS7/E1ntt9VMUjaBvgV6VLIP4Hv236os7E0m6j8QwAk7QXsZvvbzRxDo8TRCDE0ShyNEEOjxNEIMfQVA+odQAj1JukcYCdgoR/W0hdiaJQ4GiGGRomjEWJolDgaIYa+JFr+IYQQQpOJAX8hhBBCk4nKP4QQQmgyUfmHEEIITSYq/xBCCKHJROUfQgghNJn/D9Ch85CFvRqlAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<Figure size 432x288 with 2 Axes>"
      ]
     },
     "metadata": {
      "needs_background": "light"
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "#sb.heatmap(bins_df)\n",
    "import matplotlib.pyplot as plt\n",
    "ax = plt.axes()\n",
    "sb.heatmap(bins_df, ax = ax)\n",
    "ax.set_title('Species abundance over days')\n",
    "plt.show()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Question 5: Totally meta"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Enterococcus faecalis seems to dominate abundance in this graph as well and varies thoguhout the days but remains dominant. This graph seems to miss the early spike of Stapylococcus but this may be due to the representation of only one species. It did seem to pick up the large increase in Cutibacterium avidum & Anaerococcus prevotii."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
